<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="Estudios in silico en Biomedicina  Máster en Bioinformática, Universidad de Valencia" />
  <title>NGS data analysis tutorial:  Variant Calling</title>
  <style type="text/css">code{white-space: pre;}</style>
  <style>
  
  html {
      color:#000;
      background:#FFF;
  }
  
  body,div,dl,dt,dd,ul,ol,li,h1,h2,h3,h4,h5,h6,pre,code,form,fieldset,legend,input,textarea,p,blockquote,th,td {
      margin:0;
      padding:0;
  }
  
  table {
      border-spacing:0;
      font-size:inherit;
      font:100%;
      /*max-width:100%;*/
      background-color:transparent;
      border:1px solid #DDD;
      border-left:0;
      border-collapse:collapsed;
      -webkit-border-radius:4px;
      -moz-border-radius:4px;
      border-radius:4px;
  }
  
  fieldset,img {
      border:0;
  }
  
  address,caption,cite,code,dfn,em,strong,th,var {
      font-style:normal;
      font-weight:400;
  }
  
  ol,ul {
      list-style:none;
  }
  
  caption,th {
      text-align:left;
  }
  
  h1,h2,h3,h4,h5,h6 {
      font-size:100%;
      font-weight:400;
      color:#20639b;
      margin:1em 0;
  }
  
  q:before,q:after {
      content:'';
  }
  
  abbr,acronym {
      border:0;
      font-variant:normal;
      border-bottom:1px dotted #000;
      cursor:help;
  }
  
  sup {
      vertical-align:text-top;
  }
  
  sub {
      vertical-align:text-bottom;
  }
  
  input,textarea,select {
      font-family:inherit;
      font-weight:inherit;
      font-size:100%;
  }
  
  legend {
      color:#000;
  }
  
  h1,h2,h3,h4,h5,h6,strong {
      font-weight:700;
  }
  
  em {
      font-style:italic;
  }
  
  blockquote,ul,ol,dl {
      margin:1em;
      font-style:italic;
  }
  
  ol,ul,dl {
      margin-left:2em;
  }
  
  ol {
      list-style:decimal outside;
  }
  
  ul {
      list-style:disc outside;
  }
  
  dl dd {
      margin-left:1em;
  }
  
  th,td {
      line-height:1.125em;
      text-align:left;
      vertical-align:top;
      border-top:1px solid #DDD;
      border-left:1px solid #DDD;
      padding:.5em;
  }
  
  th {
      font-weight:700;
      text-align:center;
      color:#20639b;
  }
  
  table thead:first-child tr:first-child th,table tbody:first-child tr:first-child th,table tbody:first-child tr:first-child td {
      border-top:0;
  }
  
  table thead:first-child tr:first-child th:first-child,table tbody:first-child tr:first-child td:first-child {
      -webkit-border-radius:4px 0 0 0;
      -moz-border-radius:4px 0 0;
      border-radius:4px 0 0 0;
  }
  
  table thead:first-child tr:first-child th:last-child,table tbody:first-child tr:first-child td:last-child {
      -webkit-border-radius:0 4px 0 0;
      -moz-border-radius:0 4px 0 0;
      border-radius:0 4px 0 0;
  }
  
  table thead:last-child tr:last-child th:first-child,table tbody:last-child tr:last-child td:first-child {
      -webkit-border-radius:0 0 0 4px;
      -moz-border-radius:0 0 0 4px;
      border-radius:0 0 0 4px;
  }
  
  table thead:last-child tr:last-child th:last-child,table tbody:last-child tr:last-child td:last-child {
      -webkit-border-radius:0 0 4px 0;
      -moz-border-radius:0 0 4px;
      border-radius:0 0 4px 0;
  }
  
  table tbody tr:nth-child(odd) td,table tbody tr:nth-child(odd) th {
      background-color:#f9f9f9;
  }
  
  caption {
      margin-bottom:.5em;
      text-align:center;
  }
  
  p,fieldset,table,pre {
      margin-top:.5em;
      margin-bottom:.5em;
  }
  
  body {
      font:16px/1.4 "Helvetica Neue", Helvetica, "Open Sans", Arial, sans-serif;
      color:#333;
      /*max-width:960px;*/
      /*margin:0 auto;*/
      margin:1em;
  }
  
  pre,code,kbd,samp,tt {
      font-family:monospace;
      font-size:90%;
      line-height:100%;
  }
  
  h1 {
      font-size:200%;
      border-bottom:2px dotted #ddd;
  }
  
  a,a:visited,code {
      /*color:#f06024;*/
      color:#20639b;
  }
  
  a:link {text-decoration:none;}    /* unvisited link */
  a:visited {text-decoration:none;} /* visited link */
  a:hover {text-decoration:underline;}   /* mouse over link */
  a:active {text-decoration:underline;}  /* selected link */
  
  
  h2 {
      font-size:160%;
  }
  
  h3 {
      font-size:120%;
  }
  
  h4 {
      font-size:110%;
  }
  
  h5 {
      font-size:95%;
  }
  
  h6 {
      font-size:90%;
      color:#999;
  }
  
  dl {
      margin:0 0 1.125em;
  }
  
  dt {
      font-weight:700;
      line-height:1.125em;
  }
  
  dd {
      margin-left:1em;
      line-height:1.125em;
  }
  
  pre,blockquote {
      border:1px solid #ddd;
      background-color:#f1f1f1;
      -webkit-border-radius:6px;
      -moz-border-radius:6px;
      border-radius:6px;
      padding:10px;
  }
  
  blockquote {
      margin:1em 2.5em;
  }
  
  code {
      color:#20639b;
      background:#f1f1f1;
      line-height:1.5em;
  }
  
  img {
      max-width:100%;
  }
  
  ol[type=1] {
      list-style-type:decimal;
  }
  
  ol[type=a] {
      list-style-type:lower-alpha;
  }
  
  ol[type=A] {
      list-style-type:upper-alpha;
  }
  
  ol[type=I] {
      list-style-type:upper-roman;
  }
  
  ol[type=i] {
      list-style-type:lower-roman;
  }
  
  
  
  h1.title {
          font-size: 200%;
          font-weight: bold;
          padding-top: 0.2em;
          /*padding-bottom: 0.2em;*/
          text-align: center;
          border: none;
  }
  
  h2.author {
          font-size: 100%;
          text-align: center;
          border: none;
  }
  
  h3.date {
          font-size: 75%;
          text-align: center;
          border: none;
          font-style:italic;
          font-weight: bold;
  }
  
  /* animated spoiler CSS by Bloggersentral.com */
  .spoilerbutton {
  	display:block;
  	margin:5px 0;
  	padding: 3px 6px;
  	color: #20639b;
  	font-size: small;
  	font-weight: bold;
  }
  .spoiler {overflow:hidden;background: #f5f5f5;}
  .spoiler > div {-webkit-transition: all 0.2s ease;-moz-transition: margin 0.2s ease;-o-transition: all 0.2s ease;transition: margin 0.2s ease;}
  .spoilerbutton[value="Show answer"] + .spoiler > div {margin-top:-100%;}
  .spoilerbutton[value="Hide answer"] + .spoiler {padding:5px;} 
  /**********************************************/
  
  </style>
</head>
<body>
<div id="header">
<h1 class="title">NGS data analysis tutorial: <br> Variant Calling</h1>
<h2 class="author"><a href="http://www.uv.es/bioinfor/">Estudios in silico en Biomedicina</a> <br> <em>Máster en Bioinformática, Universidad de Valencia</em></h2>
<h3 class="date"><a href="http://www.dmontaner.com">David Montaner</a> <br> <em>2014-11-04</em></h3>
</div>
<hr>
<div id="TOC">
<ul>
<li><a href="#preliminaries">Preliminaries</a><ul>
<li><a href="#software-used-in-this-practical">Software used in this practical:</a></li>
<li><a href="#file-formats-explored-in-this-practical">File formats explored in this practical:</a></li>
<li><a href="#data-used-in-this-practical">Data used in this practical:</a></li>
</ul></li>
<li><a href="#overview">Overview</a></li>
<li><a href="#exercise-1">Exercise 1</a></li>
<li><a href="#exercise-2">Exercise 2</a><ul>
<li><a href="#explore-the-bam-file">Explore the BAM file</a></li>
<li><a href="#create-a-pileup-file">Create a PILEUP file</a></li>
<li><a href="#binary-format">Binary Format</a></li>
<li><a href="#create-a-vcf-file">Create a VCF file</a></li>
</ul></li>
</ul>
</div>
<hr>
<!-- Common URLs: Tools -->
<!-- Common URLs: File Formats -->
<!-- External URLs -->
<h1 id="preliminaries">Preliminaries</h1>
<h2 id="software-used-in-this-practical">Software used in this practical:</h2>
<ul>
<li><a href="http://samtools.sourceforge.net/" title="SAMtools old site">SAMtools</a> : Samtools is a suite of programs for interacting with high-throughput sequencing data; mostly with SAM/BAM. Find the manual page <a href="http://samtools.sourceforge.net/samtools.shtml" title="SAMtools online manual">here</a>.</li>
<li><a href="http://samtools.sourceforge.net/samtools.shtml#4" title="BCFtools online manual (section of the SAMtools online manual)">bcftools</a> : Part of the SAMtools suit for handling BCF files</li>
<li><a href="http://www.broadinstitute.org/igv/" title="Integrative Genomics Viewer home page">IGV</a> : Integrative Genomics Viewer</li>
<li><a href="http://sourceforge.net/projects/samstat/" title="SAMstat home page">SAMstat</a> : some statistics of the mapped reads.</li>
</ul>
<h2 id="file-formats-explored-in-this-practical">File formats explored in this practical:</h2>
<ul>
<li><a href="http://samtools.github.io/hts-specs/SAMv1.pdf" title="SAM/BAM specification">SAM/BAM</a> : Sequence Alignment/Map format. A TAB-delimited text format storing the alignment information. A <em>header section</em> is optional. BAM is the binary format.<br /> See <a href="http://samtools.sourceforge.net/samtools.shtml#5" title="SAM format summary">SAM format summary</a></li>
<li><a href="http://samtools.github.io/hts-specs/VCFv4.2.pdf" title="VCF specification">VCF</a> : Variant Calling Format<br /> See <a href="http://samtools.sourceforge.net/samtools.shtml#6" title="VCF format summary">VCF format summary</a></li>
<li><strong>pileup</strong> : Is an intermediate file format generated by <code>samtools mpileup</code> to store variant information.<br /> <a href="http://samtools.sourceforge.net/samtools.shtml#3">See samtools mpileup description</a></li>
<li><strong>BCF</strong> : binary call format a binary version of the <strong>pileup</strong> file format.</li>
</ul>
<h2 id="data-used-in-this-practical">Data used in this practical:</h2>
<ul>
<li><strong>s050_sorted.bam</strong> : the BAM file generated using <a href="http://bio-bwa.sourceforge.net/" title="Burrows-Wheeler Aligner">BWA</a> in the previous sessions. It has been sorted using <a href="http://samtools.sourceforge.net/" title="SAMtools old site">SAMtools</a>.</li>
<li><strong>f000_chr21_ref_genome_sequence.fa</strong> : the reference genome.</li>
</ul>
<h1 id="overview">Overview</h1>
<ul>
<li>We will use <a href="http://sourceforge.net/projects/samstat/" title="SAMstat home page">SAMstat</a> to get some statistics of our alignment.</li>
<li>We will use <code>samtools</code> to get a VCF file of our data.</li>
</ul>
<h1 id="exercise-1">Exercise 1</h1>
<p>Use <code>samstat</code> to get a statistical report of our reads:</p>
<pre><code>samstat s050_sorted.bam</code></pre>
<p>Explore the report generated.</p>
<ul>
<li>Does the bam file need to be sorted?</li>
<li>Try with the unsorted data.</li>
</ul>
<h1 id="exercise-2">Exercise 2</h1>
<p>Retrieve the <strong>s050_sorted.bam</strong> file generated in the previous practices.</p>
<!-- data_test directory to run my examples
    rm -r data_test
    mkdir data_test
    cp data/s050_sorted.bam data_test
    cp data/f000_chr21_ref_genome_sequence.fa data_test
-->
<pre><code>cd data</code></pre>
<h2 id="explore-the-bam-file">Explore the BAM file</h2>
<p>The header section</p>
<pre><code>samtools view -H s050_sorted.bam</code></pre>
<p>The alignment section</p>
<pre><code>samtools view s050_sorted.bam | head</code></pre>
<p>Is the file sorted?</p>
<pre><code>samtools view s050_sorted.bam | cut -f 4 | head</code></pre>
<h2 id="create-a-pileup-file">Create a PILEUP file</h2>
<p>We can use <code>samtools mpileup</code> to explore the variants in our data.</p>
<p>This command will send the information to the <em>standard output</em></p>
<pre><code>samtools mpileup -f f000_chr21_ref_genome_sequence.fa s050_sorted.bam</code></pre>
<p>So we can redirect it to a file doing</p>
<pre><code>samtools mpileup -f f000_chr21_ref_genome_sequence.fa s050_sorted.bam &gt; s060_pileup.txt</code></pre>
<ul>
<li>Why does <code>mpileup</code> need the reference genome? <!-- to know the alternative to each match or mismatch; such information is not in the sam file --></li>
<li>Why is there a <em>.fai</em> file now in the directory? <!-- the reference index has been created by samtools --></li>
<li>What would happen if the BAM file was not sorted?<br /> Try <code>samtools mpileup -f f000_chr21_ref_genome_sequence.fa s040_reads.bam</code></li>
</ul>
<p>Explore the <em>pileup</em> file generated.</p>
<pre><code>head s060_pileup.txt </code></pre>
<p>A description of the file may be found in the <a href="http://samtools.sourceforge.net/samtools.shtml#4">samtools mpileup</a> manual.</p>
<ul>
<li>Which is the first position <em>covered</em> by our sequences?</li>
<li>How does it match with the information in the bam file?</li>
<li>How many “consecutive” bases are there in the firs <em>run</em><br /> Try `<code>head -n 101 s060_pileup.txt</code></li>
<li>How did we know we had to set <code>-n 101</code> in the command above?</li>
<li>What are the meanings of the symbols in the fifth column of the <em>pileup</em> file?</li>
</ul>
<h2 id="binary-format">Binary Format</h2>
<p>The information in the <em>pileup</em> file may also be kept in a <em>binary</em> and <strong>compressed</strong> format by using the option <code>-g</code> in <code>samtools mpileup</code>.</p>
<pre><code>samtools mpileup -g -f f000_chr21_ref_genome_sequence.fa s050_sorted.bam &gt; s060_pileup.bcf</code></pre>
<ul>
<li>Can we create a binary but not compressed file?</li>
<li>When may this be useful?<br /> (See the <code>samtools mpileup</code> help)</li>
</ul>
<h2 id="create-a-vcf-file">Create a VCF file</h2>
<p>Form the BCF file we can now get a <a href="http://samtools.github.io/hts-specs/VCFv4.2.pdf" title="VCF specification">VCF</a> file:</p>
<pre><code>bcftools view s060_pileup.bcf &gt; s070_variants.vcf</code></pre>
<p>Try to concatenate the <em>mpileup</em> and the <em>bcftools</em> step</p>
<pre><code>samtools mpileup -u -f f000_chr21_ref_genome_sequence.fa s050_sorted.bam | bcftools view -&gt; my.vcf</code></pre>
<p>Use <code>diff</code> or <code>colordiff</code> to find out the differences between the two VCF files created.</p>
<ul>
<li>Why may this be useful? <!-- Because the BCF file is not really needed --></li>
</ul>
<p>Explore the VCF file created.</p>
<!-- ---------------------------------------------------------------------------

    bcftools view s060_pileup_g.bcf > s070_pileup_g.vcf
    bcftools view s060_pileup_u.bcf > s070_pileup_u.vcf

    colordiff s070_pileup_u.vcf s070_pileup_g.vcf 

each line represents a genomic position

| bcftools view bvcg

samtools mpileup [-EBug] [-C capQcoef] [-r reg] [-f in.fa] [-l list] [-M capMapQ] [-Q minBaseQ] [-q minMapQ] in.bam [in2.bam [...]]
Generate BCF or pileup for one or multiple BAM files. Alignment records are grouped by sample identifiers in @RG header lines. If sample identifiers are absent, each input file is regarded as one sample.

In the pileup format (without -u or -g),
each line represents a genomic position,
consisting of
1. chromosome name,
2. coordinate,
3. reference base,
4. read bases,
5. read qualities and
6. alignment mapping qualities.

Information on match, mismatch, indel, strand, mapping quality and start and end of a read are all encoded at the __read base column__.
At this column,
a __dot__ stands for a __match__ to the reference base on the __forward__ strand,
a __comma__ for a __match__ on the __reverse__ strand,
a ’>’ or ’<’ for a reference skip,
‘ACGTN’ for a __mismatch__ on the __forward__ strand and
‘acgtn’ for a __mismatch__ on the __reverse__ strand.
A pattern ‘\+[0-9]+[ACGTNacgtn]+’ indicates there is an __insertion__ between this reference position and the next reference position.
The length of the insertion is given by the integer in the pattern, followed by the inserted sequence.
Similarly, a pattern ‘-[0-9]+[ACGTNacgtn]+’ represents a __deletion__ from the reference.
The deleted bases will be presented as ‘*’ in the following lines. Also at the read base column, a symbol ‘^’ marks the start of a read. The ASCII of the character following ‘^’ minus 33 gives the mapping quality. A symbol ‘$’ marks the end of a read segment.


SOLO PARA LAS REGIONES DE COBERTURA

-------------------------------------------------------------------------------------------

Perdona David, se me olvidó pasarte el comando más sencillo de todos para hacer conteos con samtools:

samtools idxstats BAM > OUTPUT

el BAM debe estar ordenado con sort e indexado con index (generar el BAI)

Como resultado se obtiene un archivo tabulado con el gen, longitud, lecturas mapeadas y lecturas no mapeadas

Un saludo

-------------------------------------------------------------------------------------------

Hola David,

Te mando unos ejemplos de samttols para obtener el coverage, el coverage por región y hacer un recuento de lecturas mapeadas:

1. samtools -S -O -f REF BAM > OUTFILE (coverage por base). REF en formato fasta
2. coverageBed -split -abam BAM -b BED > OUTFILE (coverage por región, indica el número de lecturas mapeadas por región o gen). REF en  formato BED
3. samtools view -c BAM (número total de alineamientos)
4. samtools view -c -F 4 BAM (número de lecturas mapeadas)
5. samtools view -c -f 4 BAM (número de lecturas no mapeadas)
6. samtools view -c -f 1 -F12 BAM (número de lecturas mapeadas en datos paired-end)

Un saludo



-------------------------------------------------------------------------------------------

Hola David,

Te mando los comandos de samstat y el variant calling con samtools:

1. El informe con samstat se genera de una manera bastante sencilla:

samstat <BAM>

En el mismo directorio se generará un archivo HTML con el mismo nombre del BAM


2. Para realizar el variant calling con samtools:
    2.1. Hay que generar el BCF con samtools
    2.2. Covertir el BCF en VCF

outdir = variant_calling.bcf
outdir2 = variant_calling.vcf
REF = Genoma e referencia indexado (bowtie, bwa...)

samtools mpileup -uD -f REF BAM | bcftools view bvcg -> outdir

-u genera un BCF no comprimido
-D genera un output por muestra
-f genoma de referencia

bcftools view outdir | /usr/share/samtools/./vcfutils.pl varFilter -Q20 -d30 -D1000 > outdir2

-Q quality score (mapping)
-d profundidad minima
-D profundidad máxima

-->
</body>
</html>
