<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="Estudios in silico en Biomedicina  Máster en Bioinformática, Universidad de Valencia" />
  <title>NGS data analysis tutorial:  Public repositories of NGS data</title>
  <style type="text/css">code{white-space: pre;}</style>
  <style>
  
  html {
      color:#000;
      background:#FFF;
  }
  
  body,div,dl,dt,dd,ul,ol,li,h1,h2,h3,h4,h5,h6,pre,code,form,fieldset,legend,input,textarea,p,blockquote,th,td {
      margin:0;
      padding:0;
  }
  
  table {
      border-spacing:0;
      font-size:inherit;
      font:100%;
      /*max-width:100%;*/
      background-color:transparent;
      border:1px solid #DDD;
      border-left:0;
      border-collapse:collapsed;
      -webkit-border-radius:4px;
      -moz-border-radius:4px;
      border-radius:4px;
  }
  
  fieldset,img {
      border:0;
  }
  
  address,caption,cite,code,dfn,em,strong,th,var {
      font-style:normal;
      font-weight:400;
  }
  
  ol,ul {
      list-style:none;
  }
  
  caption,th {
      text-align:left;
  }
  
  h1,h2,h3,h4,h5,h6 {
      font-size:100%;
      font-weight:400;
      color:#20639b;
      margin:1em 0;
  }
  
  q:before,q:after {
      content:'';
  }
  
  abbr,acronym {
      border:0;
      font-variant:normal;
      border-bottom:1px dotted #000;
      cursor:help;
  }
  
  sup {
      vertical-align:text-top;
  }
  
  sub {
      vertical-align:text-bottom;
  }
  
  input,textarea,select {
      font-family:inherit;
      font-weight:inherit;
      font-size:100%;
  }
  
  legend {
      color:#000;
  }
  
  h1,h2,h3,h4,h5,h6,strong {
      font-weight:700;
  }
  
  em {
      font-style:italic;
  }
  
  blockquote,ul,ol,dl {
      margin:1em;
      font-style:italic;
  }
  
  ol,ul,dl {
      margin-left:2em;
  }
  
  ol {
      list-style:decimal outside;
  }
  
  ul {
      list-style:disc outside;
  }
  
  dl dd {
      margin-left:1em;
  }
  
  th,td {
      line-height:1.125em;
      text-align:left;
      vertical-align:top;
      border-top:1px solid #DDD;
      border-left:1px solid #DDD;
      padding:.5em;
  }
  
  th {
      font-weight:700;
      text-align:center;
      color:#20639b;
  }
  
  table thead:first-child tr:first-child th,table tbody:first-child tr:first-child th,table tbody:first-child tr:first-child td {
      border-top:0;
  }
  
  table thead:first-child tr:first-child th:first-child,table tbody:first-child tr:first-child td:first-child {
      -webkit-border-radius:4px 0 0 0;
      -moz-border-radius:4px 0 0;
      border-radius:4px 0 0 0;
  }
  
  table thead:first-child tr:first-child th:last-child,table tbody:first-child tr:first-child td:last-child {
      -webkit-border-radius:0 4px 0 0;
      -moz-border-radius:0 4px 0 0;
      border-radius:0 4px 0 0;
  }
  
  table thead:last-child tr:last-child th:first-child,table tbody:last-child tr:last-child td:first-child {
      -webkit-border-radius:0 0 0 4px;
      -moz-border-radius:0 0 0 4px;
      border-radius:0 0 0 4px;
  }
  
  table thead:last-child tr:last-child th:last-child,table tbody:last-child tr:last-child td:last-child {
      -webkit-border-radius:0 0 4px 0;
      -moz-border-radius:0 0 4px;
      border-radius:0 0 4px 0;
  }
  
  table tbody tr:nth-child(odd) td,table tbody tr:nth-child(odd) th {
      background-color:#f9f9f9;
  }
  
  caption {
      margin-bottom:.5em;
      text-align:center;
  }
  
  p,fieldset,table,pre {
      margin-top:.5em;
      margin-bottom:.5em;
  }
  
  body {
      font:16px/1.4 "Helvetica Neue", Helvetica, "Open Sans", Arial, sans-serif;
      color:#333;
      /*max-width:960px;*/
      /*margin:0 auto;*/
      margin:1em;
  }
  
  pre,code,kbd,samp,tt {
      font-family:monospace;
      font-size:90%;
      line-height:100%;
  }
  
  h1 {
      font-size:200%;
      border-bottom:2px dotted #ddd;
  }
  
  a,a:visited,code {
      /*color:#f06024;*/
      color:#20639b;
  }
  
  a:link {text-decoration:none;}    /* unvisited link */
  a:visited {text-decoration:none;} /* visited link */
  a:hover {text-decoration:underline;}   /* mouse over link */
  a:active {text-decoration:underline;}  /* selected link */
  
  
  h2 {
      font-size:160%;
  }
  
  h3 {
      font-size:120%;
  }
  
  h4 {
      font-size:110%;
  }
  
  h5 {
      font-size:95%;
  }
  
  h6 {
      font-size:90%;
      color:#999;
  }
  
  dl {
      margin:0 0 1.125em;
  }
  
  dt {
      font-weight:700;
      line-height:1.125em;
  }
  
  dd {
      margin-left:1em;
      line-height:1.125em;
  }
  
  pre,blockquote {
      border:1px solid #ddd;
      background-color:#f1f1f1;
      -webkit-border-radius:6px;
      -moz-border-radius:6px;
      border-radius:6px;
      padding:10px;
  }
  
  blockquote {
      margin:1em 2.5em;
  }
  
  code {
      color:#20639b;
      background:#f1f1f1;
      line-height:1.5em;
  }
  
  img {
      max-width:100%;
  }
  
  ol[type=1] {
      list-style-type:decimal;
  }
  
  ol[type=a] {
      list-style-type:lower-alpha;
  }
  
  ol[type=A] {
      list-style-type:upper-alpha;
  }
  
  ol[type=I] {
      list-style-type:upper-roman;
  }
  
  ol[type=i] {
      list-style-type:lower-roman;
  }
  
  
  
  h1.title {
          font-size: 200%;
          font-weight: bold;
          padding-top: 0.2em;
          /*padding-bottom: 0.2em;*/
          text-align: center;
          border: none;
  }
  
  h2.author {
          font-size: 100%;
          text-align: center;
          border: none;
  }
  
  h3.date {
          font-size: 75%;
          text-align: center;
          border: none;
          font-style:italic;
          font-weight: bold;
  }
  
  /* animated spoiler CSS by Bloggersentral.com */
  .spoilerbutton {
  	display:block;
  	margin:5px 0;
  	padding: 3px 6px;
  	color: #20639b;
  	font-size: small;
  	font-weight: bold;
  }
  .spoiler {overflow:hidden;background: #f5f5f5;}
  .spoiler > div {-webkit-transition: all 0.2s ease;-moz-transition: margin 0.2s ease;-o-transition: all 0.2s ease;transition: margin 0.2s ease;}
  .spoilerbutton[value="Show answer"] + .spoiler > div {margin-top:-100%;}
  .spoilerbutton[value="Hide answer"] + .spoiler {padding:5px;} 
  /**********************************************/
  
  </style>
</head>
<body>
<div id="header">
<h1 class="title">NGS data analysis tutorial: <br> Public repositories of NGS data</h1>
<h2 class="author"><a href="http://www.uv.es/bioinfor/">Estudios in silico en Biomedicina</a> <br> <em>Máster en Bioinformática, Universidad de Valencia</em></h2>
<h3 class="date"><a href="http://www.dmontaner.com">David Montaner</a> <br> <em>2014-12-16</em></h3>
</div>
<hr>
<div id="TOC">
<ul>
<li><a href="#preliminaries">Preliminaries</a><ul>
<li><a href="#software-used-in-this-practical">Software used in this practical:</a></li>
<li><a href="#file-formats-explored">File formats explored:</a></li>
</ul></li>
<li><a href="#overview">Overview</a></li>
<li><a href="#exercise">Exercise</a><ul>
<li><a href="#extract-data-in-fastq-format">Extract data in fastq format</a></li>
<li><a href="#extract-data-in-sam-format">Extract data in SAM format</a></li>
</ul></li>
</ul>
</div>
<hr>
<!-- COMMON LINKS HERE -->
<h1 id="preliminaries">Preliminaries</h1>
<p>In this practical we will download one of the samples provided in the paper from Lau et al. (2013) <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3799583" title="Lau et al. (2013)">Alteration of the microRNA network during the progression of Alzheimer’s disease</a>, and use the <a href="http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software" title="reading and writing files from the SRA database">SRA Toolkit</a> to convert it to <strong>fastq</strong> and <strong>BAM</strong> formats.</p>
<h2 id="software-used-in-this-practical">Software used in this practical:</h2>
<ul>
<li><a href="http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software" title="reading and writing files from the SRA database">SRA Toolkit</a> : tools for reading (dumping) and writing sequencing files from <a href="http://www.ncbi.nlm.nih.gov/sra/" title="Sequence Read Archive database at the NCBI">SRA</a> database.</li>
<li><a href="http://samtools.sourceforge.net/" title="SAMtools home">SAMtools</a> : utilities on manipulating alignments in the SAM format.</li>
</ul>
<p><strong>Note</strong>:</p>
<p>When installing the <a href="http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software" title="reading and writing files from the SRA database">SRA Toolkit</a> you can use the <code>md5sum</code> Shell command to verify that the binary file has been properly downloaded. Find the <em>md5 checksums</em> file in the <a href="http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software" title="reading and writing files from the SRA database">SRA Toolkit web page</a></p>
<h2 id="file-formats-explored">File formats explored:</h2>
<ul>
<li><strong>SRA</strong>: compressed binary format.</li>
<li>SAM / BAM</li>
<li>FastQ</li>
</ul>
<h1 id="overview">Overview</h1>
<ol style="list-style-type: decimal">
<li>Find trace the data location through the <a href="http://www.ncbi.nlm.nih.gov/geo/" title="Gene Expression Omnibus">GEO</a> and <a href="http://www.ncbi.nlm.nih.gov/sra/" title="Sequence Read Archive database at the NCBI">SRA</a> web sites.</li>
<li>Extract read data in <strong>fastq</strong> and <strong>SAM/BAM</strong> formats.</li>
</ol>
<h1 id="exercise">Exercise</h1>
<p>According to <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3799583" title="Lau et al. (2013)">Lau et al. (2013)</a>:</p>
<blockquote>
<p>The data obtained from the 12 libraries have been submitted<br />to the GEO database under accession number <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE48552">GSE48552</a>.</p>
</blockquote>
<p>Go to the <a href="http://www.ncbi.nlm.nih.gov/geo/" title="Gene Expression Omnibus">GEO</a> web page and find the dataset <em>(GEO series)</em>. See how the <em>Supplementary file links</em> drive you to the <a href="http://www.ncbi.nlm.nih.gov/sra/" title="Sequence Read Archive database at the NCBI">SRA</a> data base.</p>
<p>Follow the links and find the file called <strong>SRR1103937.sra</strong>.</p>
<p>Which kind of data are available in this file?</p>
<p>What does it mean the nomenclature <code>SRP/SRP026/SRP026562</code> in the GEO web page? How does it relates to the SRA FTP site?</p>
<h2 id="extract-data-in-fastq-format">Extract data in fastq format</h2>
<p>Once you have downloaded one <em>.sra</em> file you can use the <em>SRA Toolkit</em> to extract the original reads and their qualities in fastq format:</p>
<pre><code>fastq-dump SRR1103937.sra</code></pre>
<ul>
<li>How many reads are their in this file?</li>
<li>How big is the SRA file? And the FASATQ file?</li>
<li>Use <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc" title="FastQC home page">FastQC</a> to explore data quality.</li>
</ul>
<!--
wc -l SRR1103937.fastq  ## 62 639 152 SRR1103937.fastq
fastqc SRR1103937.fastq
-->
<p>But the <em>SRA Toolkit</em> also allows you reading the <em>.sra</em> file directly form the <strong>FTP</strong> site by simply using the file ID:</p>
<pre><code>fastq-dump SRR926292</code></pre>
<ul>
<li>Explore the downloading options typing <code>fastq-dump --help</code></li>
<li>Can you download compressed formats?</li>
</ul>
<h2 id="extract-data-in-sam-format">Extract data in SAM format</h2>
<p>The <em>SRA Toolkit</em> allows you to get the <em>SAM</em> file of the <em>run</em>:</p>
<pre><code>sam-dump SRR1103937.sra &gt; mysamfile.sam</code></pre>
<p>And now you can use <a href="http://samtools.sourceforge.net/" title="SAMtools home">SAMtools</a> to handle the file, for instance to convert it into a <em>BAM</em> file:</p>
<pre><code>samtools view -Sb mysamfile.sam -o mysamfile.bam</code></pre>
<p>You can perform the two previous steps at a time using the pipe operator <code>|</code>:</p>
<pre><code>sam-dump SRR1103937.sra | samtools view -Sb mysamfile.sam -o mysamfile.bam </code></pre>
<p><code>sam-dump</code> may also be used to access remote data using the file ID:</p>
<pre><code>sam-dump SRR1103937 | head </code></pre>
<ul>
<li>How many aligned reads are there in the sample? <!-- cut -f 2 mysamfile.sam | uniq  ## solo 4 --></li>
</ul>
<!-- ---------------------------------------------------------------------------
You may want _sam-dump_ to include SAM headers:

    sam-dump --header SRR1103937.sra > mysamfile_with_header.sam


Or to include your own text into the header:

    sam-dump --header-comment 'THIS IS MY CUSTOM TEXT' SRR1103937.sra > mysamfile_with_custom_header.sam

But watch out for the software the version: 

    sam-dump --version
------------------------------------------------------------------------------->
</body>
</html>
