<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="Estudios in silico en Biomedicina  Máster en Bioinformática, Universidad de Valencia" />
  <title>NGS data analysis tutorial:  Variant Calling with GATK</title>
  <style type="text/css">code{white-space: pre;}</style>
  <style>
  
  html {
      color:#000;
      background:#FFF;
  }
  
  body,div,dl,dt,dd,ul,ol,li,h1,h2,h3,h4,h5,h6,pre,code,form,fieldset,legend,input,textarea,p,blockquote,th,td {
      margin:0;
      padding:0;
  }
  
  table {
      border-spacing:0;
      font-size:inherit;
      font:100%;
      /*max-width:100%;*/
      background-color:transparent;
      border:1px solid #DDD;
      border-left:0;
      border-collapse:collapsed;
      -webkit-border-radius:4px;
      -moz-border-radius:4px;
      border-radius:4px;
  }
  
  fieldset,img {
      border:0;
  }
  
  address,caption,cite,code,dfn,em,strong,th,var {
      font-style:normal;
      font-weight:400;
  }
  
  ol,ul {
      list-style:none;
  }
  
  caption,th {
      text-align:left;
  }
  
  h1,h2,h3,h4,h5,h6 {
      font-size:100%;
      font-weight:400;
      color:#20639b;
      margin:1em 0;
  }
  
  q:before,q:after {
      content:'';
  }
  
  abbr,acronym {
      border:0;
      font-variant:normal;
      border-bottom:1px dotted #000;
      cursor:help;
  }
  
  sup {
      vertical-align:text-top;
  }
  
  sub {
      vertical-align:text-bottom;
  }
  
  input,textarea,select {
      font-family:inherit;
      font-weight:inherit;
      font-size:100%;
  }
  
  legend {
      color:#000;
  }
  
  h1,h2,h3,h4,h5,h6,strong {
      font-weight:700;
  }
  
  em {
      font-style:italic;
  }
  
  blockquote,ul,ol,dl {
      margin:1em;
      font-style:italic;
  }
  
  ol,ul,dl {
      margin-left:2em;
  }
  
  ol {
      list-style:decimal outside;
  }
  
  ul {
      list-style:disc outside;
  }
  
  dl dd {
      margin-left:1em;
  }
  
  th,td {
      line-height:1.125em;
      text-align:left;
      vertical-align:top;
      border-top:1px solid #DDD;
      border-left:1px solid #DDD;
      padding:.5em;
  }
  
  th {
      font-weight:700;
      text-align:center;
      color:#20639b;
  }
  
  table thead:first-child tr:first-child th,table tbody:first-child tr:first-child th,table tbody:first-child tr:first-child td {
      border-top:0;
  }
  
  table thead:first-child tr:first-child th:first-child,table tbody:first-child tr:first-child td:first-child {
      -webkit-border-radius:4px 0 0 0;
      -moz-border-radius:4px 0 0;
      border-radius:4px 0 0 0;
  }
  
  table thead:first-child tr:first-child th:last-child,table tbody:first-child tr:first-child td:last-child {
      -webkit-border-radius:0 4px 0 0;
      -moz-border-radius:0 4px 0 0;
      border-radius:0 4px 0 0;
  }
  
  table thead:last-child tr:last-child th:first-child,table tbody:last-child tr:last-child td:first-child {
      -webkit-border-radius:0 0 0 4px;
      -moz-border-radius:0 0 0 4px;
      border-radius:0 0 0 4px;
  }
  
  table thead:last-child tr:last-child th:last-child,table tbody:last-child tr:last-child td:last-child {
      -webkit-border-radius:0 0 4px 0;
      -moz-border-radius:0 0 4px;
      border-radius:0 0 4px 0;
  }
  
  table tbody tr:nth-child(odd) td,table tbody tr:nth-child(odd) th {
      background-color:#f9f9f9;
  }
  
  caption {
      margin-bottom:.5em;
      text-align:center;
  }
  
  p,fieldset,table,pre {
      margin-top:.5em;
      margin-bottom:.5em;
  }
  
  body {
      font:16px/1.4 "Helvetica Neue", Helvetica, "Open Sans", Arial, sans-serif;
      color:#333;
      /*max-width:960px;*/
      /*margin:0 auto;*/
      margin:1em;
  }
  
  pre,code,kbd,samp,tt {
      font-family:monospace;
      font-size:90%;
      line-height:100%;
  }
  
  h1 {
      font-size:200%;
      border-bottom:2px dotted #ddd;
  }
  
  a,a:visited,code {
      /*color:#f06024;*/
      color:#20639b;
  }
  
  a:link {text-decoration:none;}    /* unvisited link */
  a:visited {text-decoration:none;} /* visited link */
  a:hover {text-decoration:underline;}   /* mouse over link */
  a:active {text-decoration:underline;}  /* selected link */
  
  
  h2 {
      font-size:160%;
  }
  
  h3 {
      font-size:120%;
  }
  
  h4 {
      font-size:110%;
  }
  
  h5 {
      font-size:95%;
  }
  
  h6 {
      font-size:90%;
      color:#999;
  }
  
  dl {
      margin:0 0 1.125em;
  }
  
  dt {
      font-weight:700;
      line-height:1.125em;
  }
  
  dd {
      margin-left:1em;
      line-height:1.125em;
  }
  
  pre,blockquote {
      border:1px solid #ddd;
      background-color:#f1f1f1;
      -webkit-border-radius:6px;
      -moz-border-radius:6px;
      border-radius:6px;
      padding:10px;
  }
  
  blockquote {
      margin:1em 2.5em;
  }
  
  code {
      color:#20639b;
      background:#f1f1f1;
      line-height:1.5em;
  }
  
  img {
      max-width:100%;
  }
  
  ol[type=1] {
      list-style-type:decimal;
  }
  
  ol[type=a] {
      list-style-type:lower-alpha;
  }
  
  ol[type=A] {
      list-style-type:upper-alpha;
  }
  
  ol[type=I] {
      list-style-type:upper-roman;
  }
  
  ol[type=i] {
      list-style-type:lower-roman;
  }
  
  
  
  h1.title {
          font-size: 200%;
          font-weight: bold;
          padding-top: 0.2em;
          /*padding-bottom: 0.2em;*/
          text-align: center;
          border: none;
  }
  
  h2.author {
          font-size: 100%;
          text-align: center;
          border: none;
  }
  
  h3.date {
          font-size: 75%;
          text-align: center;
          border: none;
          font-style:italic;
          font-weight: bold;
  }
  
  /* animated spoiler CSS by Bloggersentral.com */
  .spoilerbutton {
  	display:block;
  	margin:5px 0;
  	padding: 3px 6px;
  	color: #20639b;
  	font-size: small;
  	font-weight: bold;
  }
  .spoiler {overflow:hidden;background: #f5f5f5;}
  .spoiler > div {-webkit-transition: all 0.2s ease;-moz-transition: margin 0.2s ease;-o-transition: all 0.2s ease;transition: margin 0.2s ease;}
  .spoilerbutton[value="Show answer"] + .spoiler > div {margin-top:-100%;}
  .spoilerbutton[value="Hide answer"] + .spoiler {padding:5px;} 
  /**********************************************/
  
  </style>
</head>
<body>
<div id="header">
<h1 class="title">NGS data analysis tutorial: <br> Variant Calling with GATK</h1>
<h2 class="author"><a href="http://www.uv.es/bioinfor/">Estudios in silico en Biomedicina</a> <br> <em>Máster en Bioinformática, Universidad de Valencia</em></h2>
<h3 class="date"><a href="http://www.dmontaner.com">David Montaner</a> <br> <em>2014-11-11</em></h3>
</div>
<hr>
<div id="TOC">
<ul>
<li><a href="#preliminaries">Preliminaries</a><ul>
<li><a href="#software-used-in-this-practical">Software used in this practical:</a></li>
<li><a href="#file-formats-explored">File formats explored:</a></li>
</ul></li>
<li><a href="#exercise-1-variant-calling-with-paired-end-data">Exercise 1: Variant calling with paired-end data</a><ul>
<li><a href="#prepare-reference-genome-generate-the-fasta-file-index">1. Prepare reference genome: generate the fasta file index</a></li>
<li><a href="#prepare-bam-file">2. Prepare BAM file</a></li>
<li><a href="#mark-duplicates-using-picard">3. Mark duplicates (using Picard)</a></li>
<li><a href="#local-realignment-around-indels-using-gatk">4. Local realignment around INDELS (using GATK)</a></li>
<li><a href="#base-quality-score-recalibration-using-gatk">5. Base quality score recalibration (using GATK)</a></li>
<li><a href="#variant-calling-using-gatk---unifiedgenotyper">6. Variant calling (using GATK - <strong>UnifiedGenotyper</strong>)</a></li>
<li><a href="#introduce-filters-in-the-vcf-file">7. Introduce filters in the VCF file</a></li>
</ul></li>
<li><a href="#exercise-2-variant-calling-with-single-end-data">Exercise 2: Variant calling with single-end data</a><ul>
<li><a href="#prepare-reference-genome-generate-the-fasta-file-index-1">1. Prepare reference genome: generate the fasta file index</a></li>
<li><a href="#prepare-bam-file-1">2. Prepare BAM file</a></li>
<li><a href="#mark-duplicates-using-picard-1">3. Mark duplicates (using Picard)</a></li>
<li><a href="#local-realignment-around-indels-using-gatk-1">4. Local realignment around INDELS (using GATK)</a></li>
<li><a href="#base-quality-score-recalibration-using-gatk-1">5. Base quality score recalibration (using GATK)</a></li>
<li><a href="#variant-calling-using-gatk---unifiedgenotyper-1">6. Variant calling (using GATK - <strong>UnifiedGenotyper</strong>)</a></li>
<li><a href="#compare-paired-end-vcf-against-single-end-vcf">7. Compare paired-end VCF against single-end VCF</a></li>
</ul></li>
</ul>
</div>
<hr>
<!-- COMMON LINKS HERE -->
<h1 id="preliminaries">Preliminaries</h1>
<h2 id="software-used-in-this-practical">Software used in this practical:</h2>
<ul>
<li><a href="http://samtools.sourceforge.net/" title="samtools">SAMTools</a> : SAM Tools provide various utilities for manipulating alignments in the SAM format, including sorting, merging, indexing and generating alignments in a per-position format.</li>
<li><a href="http://picard.sourceforge.net/" title="Picard">Picard</a> : Picard comprises Java-based command-line utilities that manipulate SAM files, and a Java API (SAM-JDK) for creating new programs that read and write SAM files.</li>
<li><a href="http://www.broadinstitute.org/gatk/" title="GATK">GATK</a> : Genome Analysis Toolkit - A package to analyse next-generation re-sequencing data, primary focused on variant discovery and genotyping.</li>
</ul>
<p><strong>Note on Picard</strong></p>
<p>In this tutorial we asume you did install Picard using <code>apt-get</code>. Then you will be able to call <code>picard-tools</code> from your shell.</p>
<p>If you have a custom installation of Picard then you will need to call it via Java doing something like:</p>
<pre><code>java -jar ../picard/CreateSequenceDictionary.jar</code></pre>
<p>instead of</p>
<pre><code>picard-tools CreateSequenceDictionary</code></pre>
<p>see <a href="https://github.com/biocosas/ngs_software_installation/blob/master/picard.md">here</a> for picard installation.</p>
<p><strong>GATK</strong></p>
<p>Depending on your installation of GATK. You may run it typing</p>
<pre><code>GATK</code></pre>
<p>or you will need to execute it via Java. Then the call will be similar to:</p>
<pre><code>java -jar ../gatk/GenomeAnalysisTK.jar</code></pre>
<p>see <a href="https://github.com/biocosas/ngs_software_installation/blob/master/gatk.md">here</a> for GATK installation.</p>
<h2 id="file-formats-explored">File formats explored:</h2>
<ul>
<li><a href="http://samtools.sourceforge.net/SAMv1.pdf">SAM</a></li>
<li><a href="http://www.broadinstitute.org/igv/bam">BAM</a></li>
<li>VCF Variant Call Format: see <a href="http://www.1000genomes.org/wiki/analysis/variant-call-format/vcf-variant-call-format-version-42">1000 Genomes</a> and <a href="http://en.wikipedia.org/wiki/Variant_Call_Format">Wikipedia</a> specifications.</li>
</ul>
<h1 id="exercise-1-variant-calling-with-paired-end-data">Exercise 1: Variant calling with paired-end data</h1>
<p>In this practical we will find variants in the paired-end BAM file <code>000-dna_chr21_100_hq_pe.bam</code>. We will need the <code>f000_chr21_ref_genome_sequence.fa</code> reference file used when mapping the reads and creating the BAM file. We will also need a file describing the <em>known variant sites</em> 000-dbSNP_chr21.vcf. This will be used in the recalibration steps.</p>
<p>These datasets contain reads only for the <strong>chromosome 21</strong>.</p>
<p>Copy the necessary data in your working directory:</p>
<pre><code>cd data</code></pre>
<!-- data_test directory to run my examples
    cd ..
    rm -r data_test
    mkdir data_test
    cp data/f000_chr21_ref_genome_sequence.fa data_test
    cp data/000-dna_chr21_100_hq_se.bam       data_test
    cp data/000-dna_chr21_100_hq_pe.bam       data_test 
    cp data/000-dbSNP_chr21.vcf               data_test
    cd data_test
-->
<h2 id="prepare-reference-genome-generate-the-fasta-file-index">1. Prepare reference genome: generate the fasta file index</h2>
<p>Use <code>SAMTools</code> to generate the fasta file index:</p>
<pre><code>samtools faidx f000_chr21_ref_genome_sequence.fa</code></pre>
<p>This creates a file called samtools faidx f000_chr21_ref_genome_sequence.fa.fai, with one record per line for each of the contigs in the FASTA reference file.</p>
<p>Generate the sequence dictionary using <code>Picard</code>:</p>
<pre><code>##java -jar ../picard/CreateSequenceDictionary.jar REFERENCE=f000_chr21_ref_genome_sequence.fa OUTPUT=f000_chr21_ref_genome_sequence.dict
picard-tools CreateSequenceDictionary REFERENCE=f000_chr21_ref_genome_sequence.fa OUTPUT=f000_chr21_ref_genome_sequence.dict</code></pre>
<h2 id="prepare-bam-file">2. Prepare BAM file</h2>
<!--
The **read group** information is key for downstream GATK functionality.
The GATK will not work without a read group tag.
Make sure to enter as much metadata as you know about your data in the read group fields provided.
For more information about all the possible fields in the @RG tag, take a look at the SAM specification.

    AddOrReplaceReadGroups.jar I=f000-dna_100_high_pe.bam O=f010-dna_100_high_pe_fixRG.bam RGID=group1 RGLB=lib1 RGPL=illumina RGSM=sample1 RGPU=unit1
-->
<p>We must sort and index the BAM file before processing it with Picard and GATK. To sort the bam file we use <code>samtools</code></p>
<pre><code>samtools sort 000-dna_chr21_100_hq_pe.bam 001-dna_chr21_100_hq_pe_sorted</code></pre>
<p>Index the BAM file:</p>
<pre><code>samtools index 001-dna_chr21_100_hq_pe_sorted.bam</code></pre>
<h2 id="mark-duplicates-using-picard">3. Mark duplicates (using Picard)</h2>
<p>Run the following <strong>Picard</strong> command to mark duplicates:</p>
<pre><code>##java -jar ../picard/MarkDuplicates.jar INPUT=001-dna_chr21_100_hq_pe_sorted.bam OUTPUT=002-dna_chr21_100_hq_pe_sorted_noDup.bam METRICS_FILE=002-metrics.txt
picard-tools MarkDuplicates INPUT=001-dna_chr21_100_hq_pe_sorted.bam OUTPUT=002-dna_chr21_100_hq_pe_sorted_noDup.bam METRICS_FILE=002-metrics.txt</code></pre>
<p>This creates a sorted BAM file called <code>002-dna_chr21_100_hq_pe_sorted_noDup.bam</code> with the same content as the input file, except that any duplicate reads are marked as such. It also produces a metrics file called <code>metrics.txt</code> containing (can you guess?) metrics.</p>
<p><strong>QUESTION:</strong> How many reads are removed as duplicates from the files (hint view the on-screen output from the two commands)?</p>
<p>Run the following <strong>Picard</strong> command to index the new BAM file:</p>
<pre><code>##java -jar ../picard/BuildBamIndex.jar INPUT=002-dna_chr21_100_hq_pe_sorted_noDup.bam
picard-tools BuildBamIndex INPUT=002-dna_chr21_100_hq_pe_sorted_noDup.bam</code></pre>
<h2 id="local-realignment-around-indels-using-gatk">4. Local realignment around INDELS (using GATK)</h2>
<p>There are 2 steps to the realignment process:</p>
<p><strong>First</strong>, create a target list of intervals which need to be realigned</p>
<pre><code>##java -jar ../gatk/GenomeAnalysisTK.jar -T RealignerTargetCreator -R f000_chr21_ref_genome_sequence.fa -I 002-dna_chr21_100_hq_pe_sorted_noDup.bam -o 003-indelRealigner.intervals
GATK -T RealignerTargetCreator -R f000_chr21_ref_genome_sequence.fa -I 002-dna_chr21_100_hq_pe_sorted_noDup.bam -o 003-indelRealigner.intervals</code></pre>
<p><strong>Second</strong>, perform realignment of the target intervals</p>
<pre><code>##java -jar ../gatk/GenomeAnalysisTK.jar -T IndelRealigner -R f000_chr21_ref_genome_sequence.fa -I 002-dna_chr21_100_hq_pe_sorted_noDup.bam -targetIntervals 003-indelRealigner.intervals -o 003-dna_chr21_100_hq_pe_sorted_noDup_realigned.bam
GATK -T IndelRealigner -R f000_chr21_ref_genome_sequence.fa -I 002-dna_chr21_100_hq_pe_sorted_noDup.bam -targetIntervals 003-indelRealigner.intervals -o 003-dna_chr21_100_hq_pe_sorted_noDup_realigned.bam</code></pre>
<p>This creates a file called <code>003-dna_chr21_100_hq_pe_sorted_noDup_realigned.bam</code> containing all the original reads, but with better local alignments in the regions that were realigned.</p>
<h2 id="base-quality-score-recalibration-using-gatk">5. Base quality score recalibration (using GATK)</h2>
<p>Two steps:</p>
<p><strong>First</strong>, analyse patterns of covariation in the sequence dataset</p>
<pre><code>##java -jar ../gatk/GenomeAnalysisTK.jar -T BaseRecalibrator -R f000_chr21_ref_genome_sequence.fa -I 003-dna_chr21_100_hq_pe_sorted_noDup_realigned.bam -knownSites ../000-dbSNP_chr21.vcf -o 004-recalibration_data.table
GATK -T BaseRecalibrator -R f000_chr21_ref_genome_sequence.fa -I 003-dna_chr21_100_hq_pe_sorted_noDup_realigned.bam -knownSites 000-dbSNP_chr21.vcf -o 004-recalibration_data.table</code></pre>
<p>This creates a GATKReport file called <code>004-recalibration_data.table</code> containing several tables. These tables contain the covariation data that will be used in a later step to recalibrate the base qualities of your sequence data.</p>
<p>It is imperative that you provide the program with a set of <strong>known sites</strong>, otherwise it will refuse to run. The known sites are used to build the covariation model and estimate empirical base qualities. For details on what to do if there are no known sites available for your organism of study, please see the online GATK documentation.</p>
<p><strong>Second</strong>, apply the recalibration to your sequence data</p>
<pre><code>##java -jar ../gatk/GenomeAnalysisTK.jar -T PrintReads -R f000_chr21_ref_genome_sequence.fa -I 003-dna_chr21_100_hq_pe_sorted_noDup_realigned.bam -BQSR 004-recalibration_data.table -o 004-dna_chr21_100_hq_pe_sorted_noDup_realigned_recalibrated.bam
GATK -T PrintReads -R f000_chr21_ref_genome_sequence.fa -I 003-dna_chr21_100_hq_pe_sorted_noDup_realigned.bam -BQSR 004-recalibration_data.table -o 004-dna_chr21_100_hq_pe_sorted_noDup_realigned_recalibrated.bam</code></pre>
<p>This creates a file called <code>004-dna_chr21_100_hq_pe_sorted_noDup_realigned_recalibrated.bam</code> containing all the original reads, but now with exquisitely accurate base substitution, insertion and deletion quality scores. By default, the original quality scores are discarded in order to keep the file size down. However, you have the option to retain them by adding the flag <code>–emit_original_quals</code> to the <code>PrintReads</code> command, in which case the original qualities will also be written in the file, tagged OQ.</p>
<h2 id="variant-calling-using-gatk---unifiedgenotyper">6. Variant calling (using GATK - <strong>UnifiedGenotyper</strong>)</h2>
<p>SNPs and INDELS are called using separate instructions.</p>
<p><strong>SNP calling</strong></p>
<pre><code>##java -jar ../gatk/GenomeAnalysisTK.jar -T UnifiedGenotyper -R f000_chr21_ref_genome_sequence.fa -I 004-dna_chr21_100_hq_pe_sorted_noDup_realigned_recalibrated.bam -glm SNP -o 005-dna_chr21_100_he_pe_snps.vcf
GATK -T UnifiedGenotyper -R f000_chr21_ref_genome_sequence.fa -I 004-dna_chr21_100_hq_pe_sorted_noDup_realigned_recalibrated.bam -glm SNP -o 005-dna_chr21_100_he_pe_snps.vcf</code></pre>
<p><strong>INDEL calling</strong></p>
<pre><code>##java -jar ../gatk/GenomeAnalysisTK.jar -T UnifiedGenotyper -R f000_chr21_ref_genome_sequence.fa -I 004-dna_chr21_100_hq_pe_sorted_noDup_realigned_recalibrated.bam -glm INDEL -o 005-dna_chr21_100_hq_pe_indel.vcf
GATK -T UnifiedGenotyper -R f000_chr21_ref_genome_sequence.fa -I 004-dna_chr21_100_hq_pe_sorted_noDup_realigned_recalibrated.bam -glm INDEL -o 005-dna_chr21_100_hq_pe_indel.vcf</code></pre>
<h2 id="introduce-filters-in-the-vcf-file">7. Introduce filters in the VCF file</h2>
<p>Example: filter SNPs with low confidence calling (QD &lt; 12.0) and flag them as “LowConf”.</p>
<pre><code>##java -jar ../gatk/GenomeAnalysisTK.jar -T VariantFiltration -R f000_chr21_ref_genome_sequence.fa -V 005-dna_chr21_100_he_pe_snps.vcf --filterExpression &quot;QD&lt;12.0&quot; --filterName &quot;LowConf&quot; -o 006-dna_chr21_100_he_pe_snps_filtered.vcf
GATK -T VariantFiltration -R f000_chr21_ref_genome_sequence.fa -V 005-dna_chr21_100_he_pe_snps.vcf --filterExpression &quot;QD&lt;12.0&quot; --filterName &quot;LowConf&quot; -o 006-dna_chr21_100_he_pe_snps_filtered.vcf</code></pre>
<p>The command <code>--filterExpression</code> will read the INFO field and check whether variants satisfy the requirement. If a variant does not satisfy your filter expression, the field FILTER will be filled with the indicated <code>--filterName</code>. These commands can be called several times indicating different filtering expression (i.e: –filterName One –filterExpression “X &lt; 1” –filterName Two –filterExpression “X &gt; 2”).</p>
<ul>
<li><p>How many “LowConf” variants have we obtained?</p>
<p>grep LowConf 006-dna_chr21_100_he_pe_snps_filtered.vcf | wc -l</p></li>
<li><p>What happen if you include some white spaces into the filter expression? &lt;!– some error is returned -&gt;</p>
<p>–filterExpression “QD &lt; 12.0”</p></li>
</ul>
<h1 id="exercise-2-variant-calling-with-single-end-data">Exercise 2: Variant calling with single-end data</h1>
<p>Go to the exercise2 folder in your course directory:</p>
<pre><code>cd /home/participant/cambridge_mda14/calling/example2</code></pre>
<h2 id="prepare-reference-genome-generate-the-fasta-file-index-1">1. Prepare reference genome: generate the fasta file index</h2>
<p>This step is no longer needed since we have already done it in <a href="http://ngs-course.github.io/Course_Materials/variant_calling/tutorial/010_example.html">example1</a></p>
<h2 id="prepare-bam-file-1">2. Prepare BAM file</h2>
<p>We must sort the BAM file using <code>samtools</code>:</p>
<pre><code>samtools sort 000-dna_chr21_100_hq_se.bam 001-dna_chr21_100_hq_se_sorted</code></pre>
<p>Index the BAM file:</p>
<pre><code>samtools index 001-dna_chr21_100_hq_se_sorted.bam</code></pre>
<h2 id="mark-duplicates-using-picard-1">3. Mark duplicates (using Picard)</h2>
<p>Mark and remove duplicates:</p>
<pre><code>java -jar ../picard/MarkDuplicates.jar INPUT=001-dna_chr21_100_hq_se_sorted.bam OUTPUT=002-dna_chr21_100_hq_se_sorted_noDup.bam METRICS_FILE=002-metrics.txt</code></pre>
<p>Index the new BAM file:</p>
<pre><code>java -jar ../picard/BuildBamIndex.jar INPUT=002-dna_chr21_100_hq_se_sorted_noDup.bam</code></pre>
<h2 id="local-realignment-around-indels-using-gatk-1">4. Local realignment around INDELS (using GATK)</h2>
<p>There are 2 steps to the realignment process:</p>
<p>Create a target list of intervals which need to be realigned</p>
<pre><code>java -jar ../gatk/GenomeAnalysisTK.jar -T RealignerTargetCreator -R f000_chr21_ref_genome_sequence.fa -I 002-dna_chr21_100_hq_se_sorted_noDup.bam -o 003-indelRealigner.intervals</code></pre>
<p>Perform realignment of the target intervals:</p>
<pre><code>java -jar ../gatk/GenomeAnalysisTK.jar -T IndelRealigner -R f000_chr21_ref_genome_sequence.fa -I 002-dna_chr21_100_hq_se_sorted_noDup.bam -targetIntervals 003-indelRealigner.intervals -o 003-dna_chr21_100_hq_se_sorted_noDup_realigned.bam</code></pre>
<h2 id="base-quality-score-recalibration-using-gatk-1">5. Base quality score recalibration (using GATK)</h2>
<p>Two steps:</p>
<p>Analyze patterns of covariation in the sequence dataset</p>
<pre><code>java -jar ../gatk/GenomeAnalysisTK.jar -T BaseRecalibrator -R f000_chr21_ref_genome_sequence.fa -I 003-dna_chr21_100_hq_se_sorted_noDup_realigned.bam -knownSites ../000-dbSNP_chr21.vcf -o 004-recalibration_data.table</code></pre>
<p>Apply the recalibration to your sequence data</p>
<pre><code>java -jar ../gatk/GenomeAnalysisTK.jar -T PrintReads -R f000_chr21_ref_genome_sequence.fa -I 003-dna_chr21_100_hq_se_sorted_noDup_realigned.bam -BQSR 004-recalibration_data.table -o 004-dna_chr21_100_hq_se_sorted_noDup_realigned_recalibrated.bam</code></pre>
<h2 id="variant-calling-using-gatk---unifiedgenotyper-1">6. Variant calling (using GATK - <strong>UnifiedGenotyper</strong>)</h2>
<p><strong>SNP calling</strong></p>
<pre><code>java -jar ../gatk/GenomeAnalysisTK.jar -T UnifiedGenotyper -R f000_chr21_ref_genome_sequence.fa -I 004-dna_chr21_100_hq_se_sorted_noDup_realigned_recalibrated.bam -glm SNP -o 005-dna_chr21_100_hq_se_snps.vcf</code></pre>
<p><strong>INDEL calling</strong></p>
<pre><code>java -jar ../gatk/GenomeAnalysisTK.jar -T UnifiedGenotyper -R f000_chr21_ref_genome_sequence.fa -I 004-dna_chr21_100_hq_se_sorted_noDup_realigned_recalibrated.bam -glm INDEL -o 005-dna_chr21_100_hq_se_indel.vcf</code></pre>
<h2 id="compare-paired-end-vcf-against-single-end-vcf">7. Compare paired-end VCF against single-end VCF</h2>
<p>Open IGV and load a the paired-end VCF we have generated in the previous tutorial (<code>005-dna_chr21_100_he_pe_snps.vcf</code>), its corresponding original BAM file (<code>001-dna_chr21_100_hq_pe_sorted.bam</code>) and the processed BAM (<code>004-dna_chr21_100_hq_pe_sorted_noDup_realigned_recalibrated.bam</code>).</p>
</body>
</html>
