<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="Estudios in silico en Biomedicina  Máster en Bioinformática, Universidad de Valencia" />
  <title>NGS data analysis tutorial:  RNA-Seq Analysis using htseq-count and edgeR</title>
  <style type="text/css">code{white-space: pre;}</style>
  <style>
  
  html {
      color:#000;
      background:#FFF;
  }
  
  body,div,dl,dt,dd,ul,ol,li,h1,h2,h3,h4,h5,h6,pre,code,form,fieldset,legend,input,textarea,p,blockquote,th,td {
      margin:0;
      padding:0;
  }
  
  table {
      border-spacing:0;
      font-size:inherit;
      font:100%;
      /*max-width:100%;*/
      background-color:transparent;
      border:1px solid #DDD;
      border-left:0;
      border-collapse:collapsed;
      -webkit-border-radius:4px;
      -moz-border-radius:4px;
      border-radius:4px;
  }
  
  fieldset,img {
      border:0;
  }
  
  address,caption,cite,code,dfn,em,strong,th,var {
      font-style:normal;
      font-weight:400;
  }
  
  ol,ul {
      list-style:none;
  }
  
  caption,th {
      text-align:left;
  }
  
  h1,h2,h3,h4,h5,h6 {
      font-size:100%;
      font-weight:400;
      color:#20639b;
      margin:1em 0;
  }
  
  q:before,q:after {
      content:'';
  }
  
  abbr,acronym {
      border:0;
      font-variant:normal;
      border-bottom:1px dotted #000;
      cursor:help;
  }
  
  sup {
      vertical-align:text-top;
  }
  
  sub {
      vertical-align:text-bottom;
  }
  
  input,textarea,select {
      font-family:inherit;
      font-weight:inherit;
      font-size:100%;
  }
  
  legend {
      color:#000;
  }
  
  h1,h2,h3,h4,h5,h6,strong {
      font-weight:700;
  }
  
  em {
      font-style:italic;
  }
  
  blockquote,ul,ol,dl {
      margin:1em;
      font-style:italic;
  }
  
  ol,ul,dl {
      margin-left:2em;
  }
  
  ol {
      list-style:decimal outside;
  }
  
  ul {
      list-style:disc outside;
  }
  
  dl dd {
      margin-left:1em;
  }
  
  th,td {
      line-height:1.125em;
      text-align:left;
      vertical-align:top;
      border-top:1px solid #DDD;
      border-left:1px solid #DDD;
      padding:.5em;
  }
  
  th {
      font-weight:700;
      text-align:center;
      color:#20639b;
  }
  
  table thead:first-child tr:first-child th,table tbody:first-child tr:first-child th,table tbody:first-child tr:first-child td {
      border-top:0;
  }
  
  table thead:first-child tr:first-child th:first-child,table tbody:first-child tr:first-child td:first-child {
      -webkit-border-radius:4px 0 0 0;
      -moz-border-radius:4px 0 0;
      border-radius:4px 0 0 0;
  }
  
  table thead:first-child tr:first-child th:last-child,table tbody:first-child tr:first-child td:last-child {
      -webkit-border-radius:0 4px 0 0;
      -moz-border-radius:0 4px 0 0;
      border-radius:0 4px 0 0;
  }
  
  table thead:last-child tr:last-child th:first-child,table tbody:last-child tr:last-child td:first-child {
      -webkit-border-radius:0 0 0 4px;
      -moz-border-radius:0 0 0 4px;
      border-radius:0 0 0 4px;
  }
  
  table thead:last-child tr:last-child th:last-child,table tbody:last-child tr:last-child td:last-child {
      -webkit-border-radius:0 0 4px 0;
      -moz-border-radius:0 0 4px;
      border-radius:0 0 4px 0;
  }
  
  table tbody tr:nth-child(odd) td,table tbody tr:nth-child(odd) th {
      background-color:#f9f9f9;
  }
  
  caption {
      margin-bottom:.5em;
      text-align:center;
  }
  
  p,fieldset,table,pre {
      margin-top:.5em;
      margin-bottom:.5em;
  }
  
  body {
      font:16px/1.4 "Helvetica Neue", Helvetica, "Open Sans", Arial, sans-serif;
      color:#333;
      /*max-width:960px;*/
      /*margin:0 auto;*/
      margin:1em;
  }
  
  pre,code,kbd,samp,tt {
      font-family:monospace;
      font-size:90%;
      line-height:100%;
  }
  
  h1 {
      font-size:200%;
      border-bottom:2px dotted #ddd;
  }
  
  a,a:visited,code {
      /*color:#f06024;*/
      color:#20639b;
  }
  
  a:link {text-decoration:none;}    /* unvisited link */
  a:visited {text-decoration:none;} /* visited link */
  a:hover {text-decoration:underline;}   /* mouse over link */
  a:active {text-decoration:underline;}  /* selected link */
  
  
  h2 {
      font-size:160%;
  }
  
  h3 {
      font-size:120%;
  }
  
  h4 {
      font-size:110%;
  }
  
  h5 {
      font-size:95%;
  }
  
  h6 {
      font-size:90%;
      color:#999;
  }
  
  dl {
      margin:0 0 1.125em;
  }
  
  dt {
      font-weight:700;
      line-height:1.125em;
  }
  
  dd {
      margin-left:1em;
      line-height:1.125em;
  }
  
  pre,blockquote {
      border:1px solid #ddd;
      background-color:#f1f1f1;
      -webkit-border-radius:6px;
      -moz-border-radius:6px;
      border-radius:6px;
      padding:10px;
  }
  
  blockquote {
      margin:1em 2.5em;
  }
  
  code {
      color:#20639b;
      background:#f1f1f1;
      line-height:1.5em;
  }
  
  img {
      max-width:100%;
  }
  
  ol[type=1] {
      list-style-type:decimal;
  }
  
  ol[type=a] {
      list-style-type:lower-alpha;
  }
  
  ol[type=A] {
      list-style-type:upper-alpha;
  }
  
  ol[type=I] {
      list-style-type:upper-roman;
  }
  
  ol[type=i] {
      list-style-type:lower-roman;
  }
  
  
  
  h1.title {
          font-size: 200%;
          font-weight: bold;
          padding-top: 0.2em;
          /*padding-bottom: 0.2em;*/
          text-align: center;
          border: none;
  }
  
  h2.author {
          font-size: 100%;
          text-align: center;
          border: none;
  }
  
  h3.date {
          font-size: 75%;
          text-align: center;
          border: none;
          font-style:italic;
          font-weight: bold;
  }
  
  /* animated spoiler CSS by Bloggersentral.com */
  .spoilerbutton {
  	display:block;
  	margin:5px 0;
  	padding: 3px 6px;
  	color: #20639b;
  	font-size: small;
  	font-weight: bold;
  }
  .spoiler {overflow:hidden;background: #f5f5f5;}
  .spoiler > div {-webkit-transition: all 0.2s ease;-moz-transition: margin 0.2s ease;-o-transition: all 0.2s ease;transition: margin 0.2s ease;}
  .spoilerbutton[value="Show answer"] + .spoiler > div {margin-top:-100%;}
  .spoilerbutton[value="Hide answer"] + .spoiler {padding:5px;} 
  /**********************************************/
  
  </style>
</head>
<body>
<div id="header">
<h1 class="title">NGS data analysis tutorial: <br> RNA-Seq Analysis using htseq-count and edgeR</h1>
<h2 class="author"><a href="http://www.uv.es/bioinfor/">Estudios in silico en Biomedicina</a> <br> <em>Máster en Bioinformática, Universidad de Valencia</em></h2>
<h3 class="date"><a href="http://www.dmontaner.com">David Montaner</a> <br> <em>2014-12-09</em></h3>
</div>
<hr>
<div id="TOC">
<ul>
<li><a href="#preliminaries">Preliminaries</a><ul>
<li><a href="#software-used-in-this-practical">Software used in this practical:</a></li>
<li><a href="#file-formats-explored">File formats explored:</a></li>
</ul></li>
<li><a href="#overview">Overview</a></li>
<li><a href="#exercise">Exercise</a><ul>
<li><a href="#map-against-the-reference-genome-using-bowtie2">Map against the reference genome using bowtie2</a></li>
<li><a href="#sort-bam-files">Sort BAM files</a></li>
<li><a href="#differential-expression-using-bioconductor-edger">Differential expression using Bioconductor edgeR</a></li>
</ul></li>
</ul>
</div>
<hr>
<!-- COMMON LINKS HERE -->
<h1 id="preliminaries">Preliminaries</h1>
<h2 id="software-used-in-this-practical">Software used in this practical:</h2>
<ul>
<li><a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml" title="Bowtie2 home page">Bowtie2</a> and <a href="http://ccb.jhu.edu/software/tophat/index.shtml" title="TopHat home page">TopHat</a> alignment tools.</li>
<li><a href="http://www-huber.embl.de/users/anders/HTSeq/doc/count.html#count" title="htseq-count manual">htseq-count</a> script based on <a href="http://www-huber.embl.de/users/anders/HTSeq/doc/overview.html" title="HTSeq home page">HTSeq</a> devised to compute the number of counts mapped to each genomic feature.</li>
<li><a href="http://bioconductor.org/packages/release/bioc/html/edgeR.html" title="edgeR at Bioconductor">edgeR</a> a <a href="http://bioconductor.org/" title="Bioconductor home page">Bioconductor</a></li>
</ul>
<h2 id="file-formats-explored">File formats explored:</h2>
<ul>
<li><a href="http://samtools.sourceforge.net/SAMv1.pdf" title="SAM/BAM formats">SAM/BAM</a></li>
<li><a href="http://www.ensembl.org/info/website/upload/gff.html" title="General Feature Format">GTF</a> : General Feature Format.</li>
</ul>
<h1 id="overview">Overview</h1>
<ol style="list-style-type: decimal">
<li>We use <a href="http://ccb.jhu.edu/software/tophat/index.shtml" title="TopHat home page">tophat</a> to align RNA-Seq reads to the reference genome.</li>
<li>We use <a href="http://www-huber.embl.de/users/anders/HTSeq/doc/count.html#count" title="htseq-count manual">htseq-count</a> to compute read counts for each gene.</li>
<li>We use <a href="http://www.broadinstitute.org/igv/" title="Integrative Genomics Viewer">IGV</a> to visualize the alignments and to understand how the counts are computed.</li>
<li>We use <a href="http://bioconductor.org/packages/release/bioc/html/edgeR.html" title="edgeR at Bioconductor">edgeR</a> to perform a differential expression analysis.</li>
</ol>
<h1 id="exercise">Exercise</h1>
<p>Create an empty directory to work in the exercise and copy or download the raw data to it. You will need to copy the <strong>reference genome</strong> the <strong>GTF annotation file</strong> and the <em>paired end</em> fastq files of the 3 samples (6 files in total).</p>
<pre><code>cd data</code></pre>
<!-- data_test directory to run my examples
    cd ..
    rm -r data_test
    mkdir data_test
    cp data/* data_test/
    cd data_test
-->
<h2 id="map-against-the-reference-genome-using-bowtie2">Map against the reference genome using bowtie2</h2>
<p>Fist we need to <strong>build an index</strong> for bowtie:</p>
<pre><code>bowtie2-build f000_chr21_ref_genome_sequence.fa f001_bowtie_index</code></pre>
<p>And now we can run the <strong>alignments</strong> for the <strong>paired end</strong> files:</p>
<pre><code>tophat2 -r 300 -o f021_case_tophat_out   f001_bowtie_index   f011_case_read1.fastq f011_case_read2.fastq
tophat2 -r 300 -o f022_case_tophat_out   f001_bowtie_index   f012_case_read1.fastq f012_case_read2.fastq
tophat2 -r 300 -o f023_case_tophat_out   f001_bowtie_index   f013_case_read1.fastq f013_case_read2.fastq

tophat2 -r 300 -o f024_cont_tophat_out   f001_bowtie_index   f014_cont_read1.fastq f014_cont_read2.fastq
tophat2 -r 300 -o f025_cont_tophat_out   f001_bowtie_index   f015_cont_read1.fastq f015_cont_read2.fastq
tophat2 -r 300 -o f026_cont_tophat_out   f001_bowtie_index   f016_cont_read1.fastq f016_cont_read2.fastq</code></pre>
<p><strong>REMARK:</strong> For <strong>paired end</strong> samples the left and the right files have to be separated using an <strong>space</strong>.<br />If separated by a <strong>coma</strong>, tophat understands the two files contain reads form the same sample sequenced in a <strong>single end</strong> protocol.</p>
<h2 id="sort-bam-files">Sort BAM files</h2>
<p>Use [SAMtools] to sort the BAM files <strong>by read names</strong></p>
<pre><code>samtools sort -n f021_case_tophat_out/accepted_hits.bam g031_case_sorted_n
samtools sort -n f022_case_tophat_out/accepted_hits.bam g032_case_sorted_n
samtools sort -n f023_case_tophat_out/accepted_hits.bam g033_case_sorted_n

samtools sort -n f024_cont_tophat_out/accepted_hits.bam g034_cont_sorted_n
samtools sort -n f025_cont_tophat_out/accepted_hits.bam g035_cont_sorted_n
samtools sort -n f026_cont_tophat_out/accepted_hits.bam g036_cont_sorted_n</code></pre>
<p>This arrangement mode is required by <a href="http://www-huber.embl.de/users/anders/HTSeq/doc/count.html#count" title="htseq-count manual">htseq-count</a>.</p>
<p>You can see how the reads are sorted by name (ie. by pairs) in the BAM file.</p>
<pre><code>samtools view g031_case_sorted_n.bam | head </code></pre>
<p>an find the original reads in the fastq file</p>
<pre><code>grep -A 1 ENST00000270112_43591_43135_1_0_0_0_2:0:0_1:0:0_10 f011_case_read1.fastq
grep -A 1 ENST00000270112_43591_43135_1_0_0_0_2:0:0_1:0:0_10 f011_case_read2.fastq</code></pre>
<ul>
<li>What does the <code>-A 1</code> option does?</li>
<li>Where the two reads named exactly the same in the original files?</li>
</ul>
<p>This arrangement of the BAM files by names is needed to use <a href="http://www-huber.embl.de/users/anders/HTSeq/doc/count.html#count" title="htseq-count manual">htseq-count</a> but is not suitable to use other tools like <a href="http://www.broadinstitute.org/igv/" title="Integrative Genomics Viewer">IGV</a> for instance. Thus sometimes we will need to use two different BAM files per sample, with a different arrangement each.</p>
<p>Use [SAMtools] to sort the BAM files <strong>by chromosome position</strong> (standard/default sorting)</p>
<pre><code>samtools sort f021_case_tophat_out/accepted_hits.bam g031_case_sorted
samtools sort f022_case_tophat_out/accepted_hits.bam g032_case_sorted
samtools sort f023_case_tophat_out/accepted_hits.bam g033_case_sorted

samtools sort f024_cont_tophat_out/accepted_hits.bam g034_cont_sorted
samtools sort f025_cont_tophat_out/accepted_hits.bam g035_cont_sorted
samtools sort f026_cont_tophat_out/accepted_hits.bam g036_cont_sorted</code></pre>
<p>And index them (this step is needed by the IGV)</p>
<pre><code>samtools index g031_case_sorted.bam
samtools index g032_case_sorted.bam
samtools index g033_case_sorted.bam

samtools index g034_cont_sorted.bam
samtools index g035_cont_sorted.bam
samtools index g036_cont_sorted.bam</code></pre>
<p>Now we can use <a href="http://www-huber.embl.de/users/anders/HTSeq/doc/count.html#count" title="htseq-count manual">htseq-count</a> to compute read counts for each sample at __gene_ level</p>
<pre><code>htseq-count --format=bam --stranded=no --type=gene g031_case_sorted_n.bam f005_chr21_genome_annotation.gtf &gt; h041_case.count
htseq-count --format=bam --stranded=no --type=gene g032_case_sorted_n.bam f005_chr21_genome_annotation.gtf &gt; h042_case.count
htseq-count --format=bam --stranded=no --type=gene g033_case_sorted_n.bam f005_chr21_genome_annotation.gtf &gt; h043_case.count

htseq-count --format=bam --stranded=no --type=gene g034_cont_sorted_n.bam f005_chr21_genome_annotation.gtf &gt; h044_cont.count
htseq-count --format=bam --stranded=no --type=gene g035_cont_sorted_n.bam f005_chr21_genome_annotation.gtf &gt; h045_cont.count
htseq-count --format=bam --stranded=no --type=gene g036_cont_sorted_n.bam f005_chr21_genome_annotation.gtf &gt; h046_cont.count</code></pre>
<p>or at <strong>exon</strong> level (the default setting)</p>
<pre><code>htseq-count --format=bam --stranded=no --type=exon g031_case_sorted_n.bam f005_chr21_genome_annotation.gtf &gt; g041_case.count
htseq-count --format=bam --stranded=no --type=exon g032_case_sorted_n.bam f005_chr21_genome_annotation.gtf &gt; g042_case.count
htseq-count --format=bam --stranded=no --type=exon g033_case_sorted_n.bam f005_chr21_genome_annotation.gtf &gt; g043_case.count

htseq-count --format=bam --stranded=no --type=exon g034_cont_sorted_n.bam f005_chr21_genome_annotation.gtf &gt; g044_cont.count
htseq-count --format=bam --stranded=no --type=exon g035_cont_sorted_n.bam f005_chr21_genome_annotation.gtf &gt; g045_cont.count
htseq-count --format=bam --stranded=no --type=exon g036_cont_sorted_n.bam f005_chr21_genome_annotation.gtf &gt; g046_cont.count</code></pre>
<p>Observe the structure of the “count” files.</p>
<pre><code>head h041_case.count 
tail h041_case.count </code></pre>
<p>Compare the <strong>gene</strong> and the <strong>exon</strong> level computations.</p>
<pre><code>head h041_case.count
head g041_case.count </code></pre>
<p>Use <a href="http://www.broadinstitute.org/igv/" title="Integrative Genomics Viewer">IGV</a> to visualize the alignments and explore how the the number of counts is computed. (remember to load the BAM files sorted by chromosome position in the IGV)</p>
<p>Explore the gene “ENSG00000141959” in sample “1_case” for instance.</p>
<ul>
<li>Which is the count number assigned to this gene by <code>htseq-count</code> at a <em>gene</em> level?</li>
<li>And at a <em>exon</em> level?</li>
<li>Why are there this differences? (expand the GTF track to see the different transcripts and exons)</li>
<li>How are paired reads accounted for?</li>
<li>in both cases, <em>gene</em> or <em>exon</em> level, counts are reported at <strong>gene</strong> level. Can you find a way to report counts at <em>exon</em> level? (read the help of <code>htseq-count</code>) <!-- htseq-count --format=bam --stranded=no --type=exon --idattr=exon_id g031_case_sorted_n.bam f005_chr21_genome_annotation.gtf | head --></li>
</ul>
<h2 id="differential-expression-using-bioconductor-edger">Differential expression using Bioconductor edgeR</h2>
<p>Now that the alignments are done and counts are computed we can use The <a href="http://bioconductor.org/" title="Bioconductor home page">Bioconductor</a> library <a href="http://bioconductor.org/packages/release/bioc/html/edgeR.html" title="edgeR at Bioconductor">edgeR</a> to perform a differential expression analysis.</p>
<p>General steps of the analysis are described in the <a href="edgeR_example.r" class="uri">edgeR_example.r</a> example.</p>
</body>
</html>
