<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="Estudios in silico en Biomedicina  Máster en Bioinformática, Universidad de Valencia" />
  <title>NGS data analysis tutorial:  TABIX</title>
  <style type="text/css">code{white-space: pre;}</style>
  <style>
  
  html {
      color:#000;
      background:#FFF;
  }
  
  body,div,dl,dt,dd,ul,ol,li,h1,h2,h3,h4,h5,h6,pre,code,form,fieldset,legend,input,textarea,p,blockquote,th,td {
      margin:0;
      padding:0;
  }
  
  table {
      border-spacing:0;
      font-size:inherit;
      font:100%;
      /*max-width:100%;*/
      background-color:transparent;
      border:1px solid #DDD;
      border-left:0;
      border-collapse:collapsed;
      -webkit-border-radius:4px;
      -moz-border-radius:4px;
      border-radius:4px;
  }
  
  fieldset,img {
      border:0;
  }
  
  address,caption,cite,code,dfn,em,strong,th,var {
      font-style:normal;
      font-weight:400;
  }
  
  ol,ul {
      list-style:none;
  }
  
  caption,th {
      text-align:left;
  }
  
  h1,h2,h3,h4,h5,h6 {
      font-size:100%;
      font-weight:400;
      color:#20639b;
      margin:1em 0;
  }
  
  q:before,q:after {
      content:'';
  }
  
  abbr,acronym {
      border:0;
      font-variant:normal;
      border-bottom:1px dotted #000;
      cursor:help;
  }
  
  sup {
      vertical-align:text-top;
  }
  
  sub {
      vertical-align:text-bottom;
  }
  
  input,textarea,select {
      font-family:inherit;
      font-weight:inherit;
      font-size:100%;
  }
  
  legend {
      color:#000;
  }
  
  h1,h2,h3,h4,h5,h6,strong {
      font-weight:700;
  }
  
  em {
      font-style:italic;
  }
  
  blockquote,ul,ol,dl {
      margin:1em;
      font-style:italic;
  }
  
  ol,ul,dl {
      margin-left:2em;
  }
  
  ol {
      list-style:decimal outside;
  }
  
  ul {
      list-style:disc outside;
  }
  
  dl dd {
      margin-left:1em;
  }
  
  th,td {
      line-height:1.125em;
      text-align:left;
      vertical-align:top;
      border-top:1px solid #DDD;
      border-left:1px solid #DDD;
      padding:.5em;
  }
  
  th {
      font-weight:700;
      text-align:center;
      color:#20639b;
  }
  
  table thead:first-child tr:first-child th,table tbody:first-child tr:first-child th,table tbody:first-child tr:first-child td {
      border-top:0;
  }
  
  table thead:first-child tr:first-child th:first-child,table tbody:first-child tr:first-child td:first-child {
      -webkit-border-radius:4px 0 0 0;
      -moz-border-radius:4px 0 0;
      border-radius:4px 0 0 0;
  }
  
  table thead:first-child tr:first-child th:last-child,table tbody:first-child tr:first-child td:last-child {
      -webkit-border-radius:0 4px 0 0;
      -moz-border-radius:0 4px 0 0;
      border-radius:0 4px 0 0;
  }
  
  table thead:last-child tr:last-child th:first-child,table tbody:last-child tr:last-child td:first-child {
      -webkit-border-radius:0 0 0 4px;
      -moz-border-radius:0 0 0 4px;
      border-radius:0 0 0 4px;
  }
  
  table thead:last-child tr:last-child th:last-child,table tbody:last-child tr:last-child td:last-child {
      -webkit-border-radius:0 0 4px 0;
      -moz-border-radius:0 0 4px;
      border-radius:0 0 4px 0;
  }
  
  table tbody tr:nth-child(odd) td,table tbody tr:nth-child(odd) th {
      background-color:#f9f9f9;
  }
  
  caption {
      margin-bottom:.5em;
      text-align:center;
  }
  
  p,fieldset,table,pre {
      margin-top:.5em;
      margin-bottom:.5em;
  }
  
  body {
      font:16px/1.4 "Helvetica Neue", Helvetica, "Open Sans", Arial, sans-serif;
      color:#333;
      /*max-width:960px;*/
      /*margin:0 auto;*/
      margin:1em;
  }
  
  pre,code,kbd,samp,tt {
      font-family:monospace;
      font-size:90%;
      line-height:100%;
  }
  
  h1 {
      font-size:200%;
      border-bottom:2px dotted #ddd;
  }
  
  a,a:visited,code {
      /*color:#f06024;*/
      color:#20639b;
  }
  
  a:link {text-decoration:none;}    /* unvisited link */
  a:visited {text-decoration:none;} /* visited link */
  a:hover {text-decoration:underline;}   /* mouse over link */
  a:active {text-decoration:underline;}  /* selected link */
  
  
  h2 {
      font-size:160%;
  }
  
  h3 {
      font-size:120%;
  }
  
  h4 {
      font-size:110%;
  }
  
  h5 {
      font-size:95%;
  }
  
  h6 {
      font-size:90%;
      color:#999;
  }
  
  dl {
      margin:0 0 1.125em;
  }
  
  dt {
      font-weight:700;
      line-height:1.125em;
  }
  
  dd {
      margin-left:1em;
      line-height:1.125em;
  }
  
  pre,blockquote {
      border:1px solid #ddd;
      background-color:#f1f1f1;
      -webkit-border-radius:6px;
      -moz-border-radius:6px;
      border-radius:6px;
      padding:10px;
  }
  
  blockquote {
      margin:1em 2.5em;
  }
  
  code {
      color:#20639b;
      background:#f1f1f1;
      line-height:1.5em;
  }
  
  img {
      max-width:100%;
  }
  
  ol[type=1] {
      list-style-type:decimal;
  }
  
  ol[type=a] {
      list-style-type:lower-alpha;
  }
  
  ol[type=A] {
      list-style-type:upper-alpha;
  }
  
  ol[type=I] {
      list-style-type:upper-roman;
  }
  
  ol[type=i] {
      list-style-type:lower-roman;
  }
  
  
  
  h1.title {
          font-size: 200%;
          font-weight: bold;
          padding-top: 0.2em;
          /*padding-bottom: 0.2em;*/
          text-align: center;
          border: none;
  }
  
  h2.author {
          font-size: 100%;
          text-align: center;
          border: none;
  }
  
  h3.date {
          font-size: 75%;
          text-align: center;
          border: none;
          font-style:italic;
          font-weight: bold;
  }
  
  /* animated spoiler CSS by Bloggersentral.com */
  .spoilerbutton {
  	display:block;
  	margin:5px 0;
  	padding: 3px 6px;
  	color: #20639b;
  	font-size: small;
  	font-weight: bold;
  }
  .spoiler {overflow:hidden;background: #f5f5f5;}
  .spoiler > div {-webkit-transition: all 0.2s ease;-moz-transition: margin 0.2s ease;-o-transition: all 0.2s ease;transition: margin 0.2s ease;}
  .spoilerbutton[value="Show answer"] + .spoiler > div {margin-top:-100%;}
  .spoilerbutton[value="Hide answer"] + .spoiler {padding:5px;} 
  /**********************************************/
  
  </style>
</head>
<body>
<div id="header">
<h1 class="title">NGS data analysis tutorial: <br> TABIX</h1>
<h2 class="author"><a href="http://www.uv.es/bioinfor/">Estudios in silico en Biomedicina</a> <br> <em>Máster en Bioinformática, Universidad de Valencia</em></h2>
<h3 class="date"><a href="http://www.dmontaner.com">David Montaner</a> <br> <em>2014-11-18</em></h3>
</div>
<hr>
<div id="TOC">
<ul>
<li><a href="#preliminaries">Preliminaries</a><ul>
<li><a href="#software-used-in-this-practical">Software used in this practical:</a></li>
<li><a href="#file-formats-explored">File formats explored:</a></li>
</ul></li>
<li><a href="#overview">Overview</a></li>
<li><a href="#exercise">Exercise</a><ul>
<li><a href="#remote-access">Remote access</a></li>
</ul></li>
</ul>
</div>
<hr>
<!-- COMMON LINKS HERE -->
<h1 id="preliminaries">Preliminaries</h1>
<h2 id="software-used-in-this-practical">Software used in this practical:</h2>
<ul>
<li><a href="http://vcftools.sourceforge.net/" title="VCFtools: A package for working with VCF files: merging, comparing, annotating ...">VCFtools</a> : A package for working with VCF files: merging, comparing, annotating …</li>
<li><a href="http://samtools.sourceforge.net/tabix.shtml" title="tabix: compress and index TAB-delimited files. Useful for handling GFF, GTF, BED and VCF files">tabix</a> : compress and index TAB-delimited files. Useful for handling GFF, GTF, BED and VCF files.</li>
</ul>
<h2 id="file-formats-explored">File formats explored:</h2>
<ul>
<li>VCF Variant Call Format: see <a href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41">1000 Genomes</a> and <a href="http://en.wikipedia.org/wiki/Variant_Call_Format">Wikipedia</a> specifications.</li>
</ul>
<h1 id="overview">Overview</h1>
<ol style="list-style-type: decimal">
<li>Use <a href="http://vcftools.sourceforge.net/" title="VCFtools: A package for working with VCF files: merging, comparing, annotating ...">VCFtools</a> (perl module) to sort VCF file.</li>
<li>Use <a href="http://samtools.sourceforge.net/tabix.shtml" title="tabix: compress and index TAB-delimited files. Useful for handling GFF, GTF, BED and VCF files">tabix</a> to index the VCF file and then to query some information from it.</li>
<li>Use <a href="http://samtools.sourceforge.net/tabix.shtml" title="tabix: compress and index TAB-delimited files. Useful for handling GFF, GTF, BED and VCF files">tabix</a> to query <strong>remote</strong> data.</li>
</ol>
<h1 id="exercise">Exercise</h1>
<p>We will fork with the VCF file <code>006-dna_chr21_100_he_pe_snps_filtered.vcf</code> created in previous practicals.</p>
<p>Copy it in your working directory and go to it.</p>
<pre><code>cd data</code></pre>
<!-- data_test directory to run my examples
    cd ..
    rm -r data_test
    mkdir data_test
    cp data/005-dna_chr21_100_hq_pe_snps.vcf  data_test
    cd data_test
-->
<!-- interesante para mas adelante
vcftools --vcf 006-dna_chr21_100_he_pe_snps_filtered.vcf --out 007-fileterd --chr 21 --from-bp 9417778 --to-bp 9439226 --recode
-->
<p>In order to use <a href="http://samtools.sourceforge.net/tabix.shtml" title="tabix: compress and index TAB-delimited files. Useful for handling GFF, GTF, BED and VCF files">tabix</a> the VCF file needs to be <em>sorted</em> according to genomic position.</p>
<p>We will first use the <a href="http://vcftools.sourceforge.net/perl_module.html" title="VCFtools Perl Module man">perl module</a> in <a href="http://vcftools.sourceforge.net/man_latest.html" title="VCFtools man">VCFtools</a> to <strong>sort</strong> our data.</p>
<pre><code>cat 006-dna_chr21_100_he_pe_snps_filtered.vcf | vcf-sort &gt; f010_sorted.vcf</code></pre>
<ul>
<li>Why do we need to use the <code>cat</code> command?</li>
<li>How would you check that the <code>vcf-sort</code> command is working?</li>
</ul>
<p>In the <a href="http://samtools.sourceforge.net/tabix.shtml" title="tabix: compress and index TAB-delimited files. Useful for handling GFF, GTF, BED and VCF files">tabix</a> pipeline we need first to <strong>compress</strong> the <em>sorted</em> VCF file.</p>
<pre><code>bgzip f010_sorted.vcf</code></pre>
<ul>
<li>What happened to the file</li>
</ul>
<p>Now we can index the <em>compressed</em> file using <code>tabix</code>.</p>
<pre><code>tabix f010_sorted.vcf.gz</code></pre>
<ul>
<li>Where is the index created?</li>
</ul>
<p>Now tabix can be used to efficiently access some regions of VCF file.</p>
<pre><code>tabix f010_sorted.vcf.gz 21:9417778-9429169
tabix f010_sorted.vcf.gz 21:48095473-48110969
tabix f010_sorted.vcf.gz 21:9417778-9429169 21:48095473-48110969</code></pre>
<ul>
<li>Where are the header lines of the VCF file?</li>
<li>Can you print them using tabix? (see help)</li>
</ul>
<p>The command <code>bgzip -d</code> may help you recovering the unzipped version of the file.</p>
<h2 id="remote-access">Remote access</h2>
<p>Tabix also allows for accessing remote files… try:</p>
<pre><code>tabix ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase1/analysis_results/integrated_call_sets/ALL.chr20.integrated_phase1_v3.20101123.snps_indels_svs.genotypes.vcf.gz 20:60479-60571</code></pre>
<!--
cp sorted.vcf scorted_bckp.vcf
bgzip -c corted_bckp.vcf > sorted2.vcf.gz
-->
</body>
</html>
