<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="Estudios in silico en Biomedicina  Máster en Bioinformática, Universidad de Valencia" />
  <title>NGS data analysis tutorial:  Association Analysis using PLINK</title>
  <style type="text/css">code{white-space: pre;}</style>
  <style>
  
  html {
      color:#000;
      background:#FFF;
  }
  
  body,div,dl,dt,dd,ul,ol,li,h1,h2,h3,h4,h5,h6,pre,code,form,fieldset,legend,input,textarea,p,blockquote,th,td {
      margin:0;
      padding:0;
  }
  
  table {
      border-spacing:0;
      font-size:inherit;
      font:100%;
      /*max-width:100%;*/
      background-color:transparent;
      border:1px solid #DDD;
      border-left:0;
      border-collapse:collapsed;
      -webkit-border-radius:4px;
      -moz-border-radius:4px;
      border-radius:4px;
  }
  
  fieldset,img {
      border:0;
  }
  
  address,caption,cite,code,dfn,em,strong,th,var {
      font-style:normal;
      font-weight:400;
  }
  
  ol,ul {
      list-style:none;
  }
  
  caption,th {
      text-align:left;
  }
  
  h1,h2,h3,h4,h5,h6 {
      font-size:100%;
      font-weight:400;
      color:#20639b;
      margin:1em 0;
  }
  
  q:before,q:after {
      content:'';
  }
  
  abbr,acronym {
      border:0;
      font-variant:normal;
      border-bottom:1px dotted #000;
      cursor:help;
  }
  
  sup {
      vertical-align:text-top;
  }
  
  sub {
      vertical-align:text-bottom;
  }
  
  input,textarea,select {
      font-family:inherit;
      font-weight:inherit;
      font-size:100%;
  }
  
  legend {
      color:#000;
  }
  
  h1,h2,h3,h4,h5,h6,strong {
      font-weight:700;
  }
  
  em {
      font-style:italic;
  }
  
  blockquote,ul,ol,dl {
      margin:1em;
      font-style:italic;
  }
  
  ol,ul,dl {
      margin-left:2em;
  }
  
  ol {
      list-style:decimal outside;
  }
  
  ul {
      list-style:disc outside;
  }
  
  dl dd {
      margin-left:1em;
  }
  
  th,td {
      line-height:1.125em;
      text-align:left;
      vertical-align:top;
      border-top:1px solid #DDD;
      border-left:1px solid #DDD;
      padding:.5em;
  }
  
  th {
      font-weight:700;
      text-align:center;
      color:#20639b;
  }
  
  table thead:first-child tr:first-child th,table tbody:first-child tr:first-child th,table tbody:first-child tr:first-child td {
      border-top:0;
  }
  
  table thead:first-child tr:first-child th:first-child,table tbody:first-child tr:first-child td:first-child {
      -webkit-border-radius:4px 0 0 0;
      -moz-border-radius:4px 0 0;
      border-radius:4px 0 0 0;
  }
  
  table thead:first-child tr:first-child th:last-child,table tbody:first-child tr:first-child td:last-child {
      -webkit-border-radius:0 4px 0 0;
      -moz-border-radius:0 4px 0 0;
      border-radius:0 4px 0 0;
  }
  
  table thead:last-child tr:last-child th:first-child,table tbody:last-child tr:last-child td:first-child {
      -webkit-border-radius:0 0 0 4px;
      -moz-border-radius:0 0 0 4px;
      border-radius:0 0 0 4px;
  }
  
  table thead:last-child tr:last-child th:last-child,table tbody:last-child tr:last-child td:last-child {
      -webkit-border-radius:0 0 4px 0;
      -moz-border-radius:0 0 4px;
      border-radius:0 0 4px 0;
  }
  
  table tbody tr:nth-child(odd) td,table tbody tr:nth-child(odd) th {
      background-color:#f9f9f9;
  }
  
  caption {
      margin-bottom:.5em;
      text-align:center;
  }
  
  p,fieldset,table,pre {
      margin-top:.5em;
      margin-bottom:.5em;
  }
  
  body {
      font:16px/1.4 "Helvetica Neue", Helvetica, "Open Sans", Arial, sans-serif;
      color:#333;
      /*max-width:960px;*/
      /*margin:0 auto;*/
      margin:1em;
  }
  
  pre,code,kbd,samp,tt {
      font-family:monospace;
      font-size:90%;
      line-height:100%;
  }
  
  h1 {
      font-size:200%;
      border-bottom:2px dotted #ddd;
  }
  
  a,a:visited,code {
      /*color:#f06024;*/
      color:#20639b;
  }
  
  a:link {text-decoration:none;}    /* unvisited link */
  a:visited {text-decoration:none;} /* visited link */
  a:hover {text-decoration:underline;}   /* mouse over link */
  a:active {text-decoration:underline;}  /* selected link */
  
  
  h2 {
      font-size:160%;
  }
  
  h3 {
      font-size:120%;
  }
  
  h4 {
      font-size:110%;
  }
  
  h5 {
      font-size:95%;
  }
  
  h6 {
      font-size:90%;
      color:#999;
  }
  
  dl {
      margin:0 0 1.125em;
  }
  
  dt {
      font-weight:700;
      line-height:1.125em;
  }
  
  dd {
      margin-left:1em;
      line-height:1.125em;
  }
  
  pre,blockquote {
      border:1px solid #ddd;
      background-color:#f1f1f1;
      -webkit-border-radius:6px;
      -moz-border-radius:6px;
      border-radius:6px;
      padding:10px;
  }
  
  blockquote {
      margin:1em 2.5em;
  }
  
  code {
      color:#20639b;
      background:#f1f1f1;
      line-height:1.5em;
  }
  
  img {
      max-width:100%;
  }
  
  ol[type=1] {
      list-style-type:decimal;
  }
  
  ol[type=a] {
      list-style-type:lower-alpha;
  }
  
  ol[type=A] {
      list-style-type:upper-alpha;
  }
  
  ol[type=I] {
      list-style-type:upper-roman;
  }
  
  ol[type=i] {
      list-style-type:lower-roman;
  }
  
  
  
  h1.title {
          font-size: 200%;
          font-weight: bold;
          padding-top: 0.2em;
          /*padding-bottom: 0.2em;*/
          text-align: center;
          border: none;
  }
  
  h2.author {
          font-size: 100%;
          text-align: center;
          border: none;
  }
  
  h3.date {
          font-size: 75%;
          text-align: center;
          border: none;
          font-style:italic;
          font-weight: bold;
  }
  
  /* animated spoiler CSS by Bloggersentral.com */
  .spoilerbutton {
  	display:block;
  	margin:5px 0;
  	padding: 3px 6px;
  	color: #20639b;
  	font-size: small;
  	font-weight: bold;
  }
  .spoiler {overflow:hidden;background: #f5f5f5;}
  .spoiler > div {-webkit-transition: all 0.2s ease;-moz-transition: margin 0.2s ease;-o-transition: all 0.2s ease;transition: margin 0.2s ease;}
  .spoilerbutton[value="Show answer"] + .spoiler > div {margin-top:-100%;}
  .spoilerbutton[value="Hide answer"] + .spoiler {padding:5px;} 
  /**********************************************/
  
  </style>
</head>
<body>
<div id="header">
<h1 class="title">NGS data analysis tutorial: <br> Association Analysis using PLINK</h1>
<h2 class="author"><a href="http://www.uv.es/bioinfor/">Estudios in silico en Biomedicina</a> <br> <em>Máster en Bioinformática, Universidad de Valencia</em></h2>
<h3 class="date"><a href="http://www.dmontaner.com">David Montaner</a> <br> <em>2014-11-18</em></h3>
</div>
<hr>
<div id="TOC">
<ul>
<li><a href="#preliminaries">Preliminaries</a><ul>
<li><a href="#software-used-in-this-practical">Software used in this practical:</a></li>
<li><a href="#file-formats-explored">File formats explored:</a><ul>
<li><a href="#standard-plink-file-formats-are">Standard PLINK file formats are:</a></li>
<li><a href="#transposed-formats-generally-more-suitable-for-genomic-data-are">Transposed formats, generally more suitable for genomic data are:</a></li>
</ul></li>
</ul></li>
<li><a href="#overview">Overview</a></li>
<li><a href="#exercise">Exercise</a></li>
</ul>
</div>
<hr>
<!-- COMMON LINKS HERE -->
<h1 id="preliminaries">Preliminaries</h1>
<h2 id="software-used-in-this-practical">Software used in this practical:</h2>
<ul>
<li><a href="http://pngu.mgh.harvard.edu/~purcell/plink/" title="PLINK: whole genome association analysis">PLINK</a> : whole genome association analysis.</li>
<li><a href="http://vcftools.sourceforge.net/" title="VCFtools: A package for working with VCF files: merging, comparing, annotating ...">VCFtools</a> : A package for working with VCF files: merging, comparing, annotating …</li>
<li><a href="http://samtools.sourceforge.net/tabix.shtml" title="tabix: compress and index TAB-delimited files. Useful for handling GFF, GTF, BED and VCF files">tabix</a> : compress and index TAB-delimited files. Useful for handling GFF, GTF, BED and VCF files.</li>
<li><a href="http://www.r-project.org/" title="The Project for Statistical Computing">R</a> : a programming language devised for statistical data analysis.</li>
</ul>
<h2 id="file-formats-explored">File formats explored:</h2>
<ul>
<li>VCF Variant Call Format: see <a href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41">1000 Genomes</a> and <a href="http://en.wikipedia.org/wiki/Variant_Call_Format">Wikipedia</a> specifications.</li>
</ul>
<p>See PLINK file formats description <a href="http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml#tr">here</a>.</p>
<h3 id="standard-plink-file-formats-are">Standard PLINK file formats are:</h3>
<p><strong>PED file</strong> with fields (columns):</p>
<ol style="list-style-type: decimal">
<li>Family ID</li>
<li>Individual ID</li>
<li>Paternal ID</li>
<li>Maternal ID</li>
<li>Sex (1=male; 2=female; other=unknown)</li>
<li>Phenotype (1=unaffected; 2=affected; 0 missing; -9=missing)</li>
<li>… genotypes …</li>
</ol>
<p><strong>MAP file</strong> with fields (columns):</p>
<ol style="list-style-type: decimal">
<li>chromosome (1-22, X, Y or 0 if unplaced)</li>
<li>rs… or SNP identifier</li>
<li>Genetic distance (Morgans)</li>
<li>Base-pair position (BP units)</li>
</ol>
<h3 id="transposed-formats-generally-more-suitable-for-genomic-data-are">Transposed formats, generally more suitable for genomic data are:</h3>
<p><strong>TPED file with fields (columns):</strong></p>
<ol style="list-style-type: decimal">
<li>chromosome (1-22, X, Y or 0 if unplaced)</li>
<li>SNP identifier (rs…)</li>
<li>Genetic distance (Morgans)</li>
<li>Base-pair position (BP units)</li>
<li>… genotypes …</li>
</ol>
<p><strong>TFAM file with fields (columns):</strong></p>
<ol style="list-style-type: decimal">
<li>Family ID</li>
<li>Individual ID</li>
<li>Paternal ID</li>
<li>Maternal ID</li>
<li>Sex (1=male; 2=female; other=unknown)</li>
<li>Phenotype (1=unaffected; 2=affected; 0 missing; -9=missing)</li>
</ol>
<h1 id="overview">Overview</h1>
<ol style="list-style-type: decimal">
<li>Use <a href="http://samtools.sourceforge.net/tabix.shtml" title="tabix: compress and index TAB-delimited files. Useful for handling GFF, GTF, BED and VCF files">tabix</a> to download data from <strong>remote</strong> VCF files.</li>
<li>Use <a href="http://vcftools.sourceforge.net/" title="VCFtools: A package for working with VCF files: merging, comparing, annotating ...">VCFtools</a> for converting VCF files to PED and MAP format.</li>
<li>Use <a href="http://pngu.mgh.harvard.edu/~purcell/plink/" title="PLINK: whole genome association analysis">PLINK</a> to perform association studies.</li>
</ol>
<p>Use <a href="http://www.r-project.org/" title="The Project for Statistical Computing">R</a> for data handling and transformation.</p>
<h1 id="exercise">Exercise</h1>
<!-- new and clean data directory in the sandbox
    #rm -r ../../../../sandbox/association_studies/
    #mkdir ../../../../sandbox/association_studies/
    #cd    ../../../../sandbox/association_studies/
-->
<p>For this practical we will download an example dataset from the <a href="http://www.1000genomes.org/" title="1000 Genomes Home Page">1000 Genomes</a> web page using <a href="http://samtools.sourceforge.net/tabix.shtml" title="tabix: compress and index TAB-delimited files. Useful for handling GFF, GTF, BED and VCF files">tabix</a>.</p>
<p>Go to the 1000 Genomes FTP site:</p>
<p><a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase1/analysis_results/integrated_call_sets" class="uri">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase1/analysis_results/integrated_call_sets</a></p>
<p>The files in this directory represent the final variant call set associated with the phase1 analysis.</p>
<p>Set up your working directory:</p>
<!--
    rm -r data
-->
<pre><code>mkdir data
cd data</code></pre>
<p>Use <a href="http://samtools.sourceforge.net/tabix.shtml" title="tabix: compress and index TAB-delimited files. Useful for handling GFF, GTF, BED and VCF files">tabix</a> to download the first variants in chromosome 20:</p>
<pre><code>tabix -h ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase1/analysis_results/integrated_call_sets/ALL.chr20.integrated_phase1_v3.20101123.snps_indels_svs.genotypes.vcf.gz 20:1-1000000 &gt; f010_first_chr20.vcf</code></pre>
<ul>
<li>What does it mean the parameter <code>-h</code> in the command above? <!-- print / include also the header lines --><br /> (you can find the <code>tabix</code> help just by typing the command in in shell)</li>
<li>What does it mean the parameter <code>20:1-1000000</code>? <!-- the range of positions to be downloaded --></li>
<li>Why do not we use the option <code>-o</code> to create the output file? How do we do it instead? <!-- the option -o is not implemented; we use the redirection > instead --></li>
</ul>
<p>How many lines has it got the downloaded file:</p>
<pre><code>wc -l f010_first_chr20.vcf </code></pre>
<p>Convert VCF file to a TPED + TFAM files using <a href="http://vcftools.sourceforge.net/" title="VCFtools: A package for working with VCF files: merging, comparing, annotating ...">VCFtools</a>:</p>
<pre><code>vcftools --vcf f010_first_chr20.vcf --out f020_plink_format --plink-tped </code></pre>
<p>The above command will create the the <code>.tped</code> and <code>.tfam</code> files. Notice that it will also create an <em>index</em> for the vcf file.</p>
<p>Explore the structure of the files created.</p>
<pre><code>head f020_plink_format.tfam
head f020_plink_format.tped</code></pre>
<p>Apply <a href="http://pngu.mgh.harvard.edu/~purcell/plink/" title="PLINK: whole genome association analysis">PLINK</a> to get a description of the dataset:</p>
<pre><code>plink --noweb --allow-no-sex --out f030_plink_description --tfile f020_plink_format</code></pre>
<ul>
<li>What happens if you do not include the <code>--allow-no-sex</code> flag? <a href="#fn1" class="footnoteRef" id="fnref1"><sup>1</sup></a><br /> (See the indications in the output or in the <code>.log</code> file)</li>
</ul>
<!-- What information is in the file `.nosex`? -->
<p>Edit the file <code>f020_plink_format.tfam</code> to include some Phenotype information. Remember this information is to be coded in <strong>column 6</strong> as: 1=unaffected; 2=affected; 0 missing.</p>
<p>You could use any software, any spreadsheet for instance, to include this information. For this practical we will use <a href="http://www.r-project.org/" title="The Project for Statistical Computing">R</a></p>
<p>(The file <strong><a href="edit_tfam_file.r" class="uri">edit_tfam_file.r</a></strong> shows how to carry out the edition of the file using R)</p>
<!--
    R CMD BATCH --vanilla ../../ngs-course.github.io/Course_Materials/association_studies/tutorial/edit_tfam_file.r
-->
<p>Let’s carry out an <a href="http://pngu.mgh.harvard.edu/~purcell/plink/anal.shtml#cc">association test</a> now using PLINK.</p>
<pre><code>plink --noweb --allow-no-sex --tped f020_plink_format.tped --tfam f040_plink_format.tfam --out f050_res_plink --assoc</code></pre>
<ul>
<li>Why do we use the parameters <code>--tped</code> and <code>--tfam</code> instead of the <code>--tfile</code> used before?.</li>
<li>What happens if you do not include the <code>--allow-no-sex</code> flag?<br /> (Run the same command without this parameter and compare the output files) <!-- all statistics are returned as missing NA because all individuals are excluded from the analysis as they do not have the sex information --></li>
</ul>
<!-- without the --allow-no-sex
    plink --noweb --tped f020_plink_format.tped --tfam f040_plink_format.tfam --out f050_res_plink_NO_NO-SEX --assoc
-->
<p>See PLINK documentation on some other options to <a href="http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml#pheno">handle Phenotype data</a></p>
<p>Explore the results file:</p>
<pre><code>head f050_res_plink.assoc</code></pre>
<p>We can ask PLINK to perform a [p-value adjustment] using the option <code>--adjust</code></p>
<pre><code>plink --noweb --allow-no-sex --tped f020_plink_format.tped --tfam f040_plink_format.tfam --out f060_res_plink --assoc --adjust</code></pre>
<p>Explore the results file:</p>
<pre><code>head f060_res_plink.assoc.adjusted</code></pre>
<p>and compare with the previous file:</p>
<pre><code>head f050_res_plink.assoc</code></pre>
<p>You can use the <a href="http://www.r-project.org/" title="The Project for Statistical Computing">R</a> scripts:</p>
<ul>
<li><strong><a href="explore_plink_results.r" class="uri">explore_plink_results.r</a></strong></li>
<li><strong><a href="explore_plink_pvalues.r" class="uri">explore_plink_pvalues.r</a></strong></li>
</ul>
<p>to do the exploration and the comparison.</p>
<!--
    R CMD BATCH --vanilla ../../ngs-course.github.io/Course_Materials/association_studies/tutorial/explore_plink_results.r
    R CMD BATCH --vanilla ../../ngs-course.github.io/Course_Materials/association_studies/tutorial/explore_plink_pvalues.r
-->
<div class="footnotes">
<hr />
<ol>
<li id="fn1"><p>Warning, found 1092 individuals with ambiguous sex codes<br />These individuals will be set to missing ( or use –allow-no-sex )<br />Writing list of these individuals to [ f030_plink_description2.nosex ]<a href="#fnref1">↩</a></p></li>
</ol>
</div>
</body>
</html>
